// react-jsx-highmaps/dist/es/components/HighchartsMapChart/HighchartsMapChart.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { BaseChart } from 'react-jsx-highcharts';

var HighchartsMapChart =
/*#__PURE__*/
function (_Component) {
  _inherits(HighchartsMapChart, _Component);

  function HighchartsMapChart() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, HighchartsMapChart);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(HighchartsMapChart)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "getGeoJSON", function (map) {
      if (!map) return;
      return typeof map === 'string' ? _this.props.getHighcharts().maps[map] : map;
    });

    _defineProperty(_assertThisInitialized(_this), "callback", function (chart) {
      var geojson = _this.geojson;

      if (geojson) {
        var format = _this.props.getHighcharts().format;

        var _chart$options$credit = chart.options.credits,
            mapText = _chart$options$credit.mapText,
            mapTextFull = _chart$options$credit.mapTextFull;
        chart.mapCredits = format(mapText, {
          geojson: geojson
        });
        chart.mapCreditsFull = format(mapTextFull, {
          geojson: geojson
        });
      }

      _this.props.callback(chart);
    });

    return _this;
  }

  _createClass(HighchartsMapChart, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          map = _this$props.map,
          chart = _this$props.chart,
          rest = _objectWithoutProperties(_this$props, ["map", "chart"]);

      this.geojson = this.getGeoJSON(map);
      return React.createElement(BaseChart, _extends({
        chart: _objectSpread({}, chart, {
          map: this.geojson
        }),
        mapNavigation: {
          enabled: false
        },
        xAxis: {
          id: 'xAxis'
        },
        yAxis: {
          id: 'yAxis'
        }
      }, rest, {
        callback: this.callback,
        chartCreationFunc: this.props.getHighcharts().mapChart,
        chartType: "mapChart"
      }));
    }
  }]);

  return HighchartsMapChart;
}(Component);

_defineProperty(HighchartsMapChart, "defaultProps", {
  callback: function callback() {}
});

export default HighchartsMapChart;
