// igniteui-react-core/esm2015/lib/GridColumnWidthChangedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let GridColumnWidthChangedEventArgsDescription = /*@__PURE__*/ (() => {
    class GridColumnWidthChangedEventArgsDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.g = null;
            this.f = null;
        }
        get_type() {
            return "GridColumnWidthChangedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get column() {
            return this.j;
        }
        set column(a) {
            this.j = a;
            this.e("Column");
        }
        get oldWidth() {
            return this.g;
        }
        set oldWidth(a) {
            this.g = a;
            this.e("OldWidth");
        }
        get newWidth() {
            return this.f;
        }
        set newWidth(a) {
            this.f = a;
            this.e("NewWidth");
        }
    }
    GridColumnWidthChangedEventArgsDescription.$t = /*@__PURE__*/ markType(GridColumnWidthChangedEventArgsDescription, 'GridColumnWidthChangedEventArgsDescription', Description.$);
    return GridColumnWidthChangedEventArgsDescription;
})();
