// igniteui-react-core/esm2015/lib/WebActionStripDescription_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebDisplayDensityBaseElementDescription } from "./WebDisplayDensityBaseElementDescription";
import { markType } from "./type";
import { Description } from "./Description";
/**
 * @hidden
 */
export let WebActionStripDescription = /*@__PURE__*/ (() => {
    class WebActionStripDescription extends WebDisplayDensityBaseElementDescription {
        constructor() {
            super();
            this.q = null;
            this.k = null;
            this.o = false;
            this.m = null;
            this.s = null;
        }
        get_type() {
            return "WebActionStrip";
        }
        get context() {
            return this.q;
        }
        set context(a) {
            this.q = a;
            this.e("Context");
        }
        get actionButtons() {
            return this.k;
        }
        set actionButtons(a) {
            this.k = a;
            this.e("ActionButtons");
        }
        get hidden() {
            return this.o;
        }
        set hidden(a) {
            this.o = a;
            this.e("Hidden");
        }
        get resourceStrings() {
            return this.m;
        }
        set resourceStrings(a) {
            this.m = a;
            this.e("ResourceStrings");
        }
        get name() {
            return this.s;
        }
        set name(a) {
            this.s = a;
            this.e("Name");
        }
    }
    WebActionStripDescription.$t = /*@__PURE__*/ markType(WebActionStripDescription, 'WebActionStripDescription', WebDisplayDensityBaseElementDescription.$);
    return WebActionStripDescription;
})();
/**
 * @hidden
 */
export let WebGridActionsBaseDirectiveDescription = /*@__PURE__*/ (() => {
    class WebGridActionsBaseDirectiveDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.j = false;
            this.h = null;
            this.l = null;
        }
        get_type() {
            return "WebGridActionsBaseDirective";
        }
        get type() {
            return this.get_type();
        }
        get buttons() {
            return this.f;
        }
        set buttons(a) {
            this.f = a;
            this.e("Buttons");
        }
        get asMenuItems() {
            return this.j;
        }
        set asMenuItems(a) {
            this.j = a;
            this.e("AsMenuItems");
        }
        get strip() {
            return this.h;
        }
        set strip(a) {
            this.h = a;
            this.e("Strip");
        }
        get name() {
            return this.l;
        }
        set name(a) {
            this.l = a;
            this.e("Name");
        }
    }
    WebGridActionsBaseDirectiveDescription.$t = /*@__PURE__*/ markType(WebGridActionsBaseDirectiveDescription, 'WebGridActionsBaseDirectiveDescription', Description.$);
    return WebGridActionsBaseDirectiveDescription;
})();
