// igniteui-react-core/esm2015/lib/WebBaseExporterDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebBaseExporterDescription = /*@__PURE__*/ (() => {
    class WebBaseExporterDescription extends Description {
        constructor() {
            super();
            this.g = null;
            this.h = null;
            this.f = null;
        }
        get_type() {
            return "WebBaseExporter";
        }
        get type() {
            return this.get_type();
        }
        get exportEndedRef() {
            return this.g;
        }
        set exportEndedRef(a) {
            this.g = a;
            this.e("ExportEndedRef");
        }
        get rowExportingRef() {
            return this.h;
        }
        set rowExportingRef(a) {
            this.h = a;
            this.e("RowExportingRef");
        }
        get columnExportingRef() {
            return this.f;
        }
        set columnExportingRef(a) {
            this.f = a;
            this.e("ColumnExportingRef");
        }
    }
    WebBaseExporterDescription.$t = /*@__PURE__*/ markType(WebBaseExporterDescription, 'WebBaseExporterDescription', Description.$);
    return WebBaseExporterDescription;
})();
