// igniteui-react-core/esm2015/lib/WebColumnToggledEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebColumnToggledEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebColumnToggledEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.k = null;
            this.n = false;
        }
        get_type() {
            return "WebColumnToggledEventArgsDetail";
        }
        get column() {
            return this.k;
        }
        set column(a) {
            this.k = a;
            this.e("Column");
        }
        get checked() {
            return this.n;
        }
        set checked(a) {
            this.n = a;
            this.e("Checked");
        }
    }
    WebColumnToggledEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebColumnToggledEventArgsDetailDescription, 'WebColumnToggledEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebColumnToggledEventArgsDetailDescription.__marshalByValue1 = true;
    WebColumnToggledEventArgsDetailDescription.__marshalByValueAlias1 = "ColumnToggledEventArgsDetail";
    return WebColumnToggledEventArgsDetailDescription;
})();
