// igniteui-react-core/esm2015/lib/WebColumnMovingEndEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebColumnMovingEndEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebColumnMovingEndEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.k = null;
            this.l = null;
            this.p = false;
        }
        get_type() {
            return "WebColumnMovingEndEventArgsDetail";
        }
        get source() {
            return this.k;
        }
        set source(a) {
            this.k = a;
            this.e("Source");
        }
        get target() {
            return this.l;
        }
        set target(a) {
            this.l = a;
            this.e("Target");
        }
        get cancel() {
            return this.p;
        }
        set cancel(a) {
            this.p = a;
            this.e("Cancel");
        }
    }
    WebColumnMovingEndEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebColumnMovingEndEventArgsDetailDescription, 'WebColumnMovingEndEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebColumnMovingEndEventArgsDetailDescription.__marshalByValue1 = true;
    WebColumnMovingEndEventArgsDetailDescription.__marshalByValueAlias1 = "ColumnMovingEndEventArgsDetail";
    return WebColumnMovingEndEventArgsDetailDescription;
})();
