// igniteui-react-core/esm2015/lib/WebDisplayDensityBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebDisplayDensityBaseDescription = /*@__PURE__*/ (() => {
    class WebDisplayDensityBaseDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.g = null;
        }
        get_type() {
            return "WebDisplayDensityBase";
        }
        get type() {
            return this.get_type();
        }
        get displayDensity() {
            return this.f;
        }
        set displayDensity(a) {
            this.f = a;
            this.e("DisplayDensity");
        }
        get onDensityChangedRef() {
            return this.g;
        }
        set onDensityChangedRef(a) {
            this.g = a;
            this.e("OnDensityChangedRef");
        }
    }
    WebDisplayDensityBaseDescription.$t = /*@__PURE__*/ markType(WebDisplayDensityBaseDescription, 'WebDisplayDensityBaseDescription', Description.$);
    return WebDisplayDensityBaseDescription;
})();
