// igniteui-react-core/esm2015/lib/WebDensityChangedEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebDensityChangedEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebDensityChangedEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.n = null;
            this.m = null;
        }
        get_type() {
            return "WebDensityChangedEventArgsDetail";
        }
        get oldDensity() {
            return this.n;
        }
        set oldDensity(a) {
            this.n = a;
            this.e("OldDensity");
        }
        get newDensity() {
            return this.m;
        }
        set newDensity(a) {
            this.m = a;
            this.e("NewDensity");
        }
    }
    WebDensityChangedEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebDensityChangedEventArgsDetailDescription, 'WebDensityChangedEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebDensityChangedEventArgsDetailDescription.__marshalByValue1 = true;
    WebDensityChangedEventArgsDetailDescription.__marshalByValueAlias1 = "DensityChangedEventArgsDetail";
    return WebDensityChangedEventArgsDetailDescription;
})();
