// igniteui-react-core/esm2015/lib/WebExporterEventDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebExporterEventDescription = /*@__PURE__*/ (() => {
    class WebExporterEventDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.h = null;
            this.j = null;
            this.l = false;
        }
        get_type() {
            return "WebExporterEvent";
        }
        get type() {
            return this.get_type();
        }
        get exporter() {
            return this.f;
        }
        set exporter(a) {
            this.f = a;
            this.e("Exporter");
        }
        get options() {
            return this.h;
        }
        set options(a) {
            this.h = a;
            this.e("Options");
        }
        get grid() {
            return this.j;
        }
        set grid(a) {
            this.j = a;
            this.e("Grid");
        }
        get cancel() {
            return this.l;
        }
        set cancel(a) {
            this.l = a;
            this.e("Cancel");
        }
    }
    WebExporterEventDescription.$t = /*@__PURE__*/ markType(WebExporterEventDescription, 'WebExporterEventDescription', Description.$);
    return WebExporterEventDescription;
})();
