// igniteui-react-core/esm2015/lib/WebForOfStateDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebForOfStateDetailDescription = /*@__PURE__*/ (() => {
    class WebForOfStateDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.m = 0;
            this.l = 0;
        }
        get_type() {
            return "WebForOfStateDetail";
        }
        get startIndex() {
            return this.m;
        }
        set startIndex(a) {
            this.m = a;
            this.e("StartIndex");
        }
        get chunkSize() {
            return this.l;
        }
        set chunkSize(a) {
            this.l = a;
            this.e("ChunkSize");
        }
    }
    WebForOfStateDetailDescription.$t = /*@__PURE__*/ markType(WebForOfStateDetailDescription, 'WebForOfStateDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebForOfStateDetailDescription.__marshalByValue1 = true;
    WebForOfStateDetailDescription.__marshalByValueAlias1 = "ForOfStateDetail";
    return WebForOfStateDetailDescription;
})();
