// igniteui-react-core/esm2015/lib/WebGridActionButtonDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGridActionButtonDescription = /*@__PURE__*/ (() => {
    class WebGridActionButtonDescription extends Description {
        constructor() {
            super();
            this.f = false;
            this.i = null;
            this.h = null;
            this.j = null;
            this.k = null;
        }
        get_type() {
            return "WebGridActionButton";
        }
        get type() {
            return this.get_type();
        }
        get asMenuItem() {
            return this.f;
        }
        set asMenuItem(a) {
            this.f = a;
            this.e("AsMenuItem");
        }
        get iconName() {
            return this.i;
        }
        set iconName(a) {
            this.i = a;
            this.e("IconName");
        }
        get classNames() {
            return this.h;
        }
        set classNames(a) {
            this.h = a;
            this.e("ClassNames");
        }
        get iconSet() {
            return this.j;
        }
        set iconSet(a) {
            this.j = a;
            this.e("IconSet");
        }
        get labelText() {
            return this.k;
        }
        set labelText(a) {
            this.k = a;
            this.e("LabelText");
        }
    }
    WebGridActionButtonDescription.$t = /*@__PURE__*/ markType(WebGridActionButtonDescription, 'WebGridActionButtonDescription', Description.$);
    return WebGridActionButtonDescription;
})();
