// igniteui-react-core/esm2015/lib/WebGridResourceStringsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGridResourceStringsDescription = /*@__PURE__*/ (() => {
    class WebGridResourceStringsDescription extends Description {
        constructor() {
            super();
            this.cl = null;
            this.cm = null;
            this.ck = null;
            this.af = null;
            this.ag = null;
            this.bf = null;
            this.b9 = null;
            this.cc = null;
            this.cb = null;
            this.b8 = null;
            this.ca = null;
            this.cd = null;
            this.b6 = null;
            this.ad = null;
            this.b7 = null;
            this.bn = null;
            this.bo = null;
            this.ce = null;
            this.br = null;
            this.bs = null;
            this.bp = null;
            this.bq = null;
            this.b3 = null;
            this.b5 = null;
            this.b4 = null;
            this.bl = null;
            this.bg = null;
            this.bi = null;
            this.b2 = null;
            this.bk = null;
            this.bj = null;
            this.ch = null;
            this.cj = null;
            this.cf = null;
            this.bw = null;
            this.b0 = null;
            this.cg = null;
            this.bx = null;
            this.b1 = null;
            this.bu = null;
            this.by = null;
            this.bv = null;
            this.bz = null;
            this.ci = null;
            this.bt = null;
            this.bh = null;
            this.bm = null;
            this.dt = null;
            this.dx = null;
            this.dw = null;
            this.dy = null;
            this.ds = null;
            this.du = null;
            this.dv = null;
            this.av = null;
            this.aw = null;
            this.ax = null;
            this.ay = null;
            this.au = null;
            this.az = null;
            this.a0 = null;
            this.a1 = null;
            this.a2 = null;
            this.a3 = null;
            this.at = null;
            this.an = null;
            this.ao = null;
            this.ap = null;
            this.al = null;
            this.ai = null;
            this.a8 = null;
            this.ba = null;
            this.bb = null;
            this.ah = null;
            this.aj = null;
            this.a4 = null;
            this.bc = null;
            this.a7 = null;
            this.be = null;
            this.a9 = null;
            this.as = null;
            this.bd = null;
            this.a6 = null;
            this.ar = null;
            this.ak = null;
            this.am = null;
            this.aq = null;
            this.a5 = null;
            this.aa = null;
            this.p = null;
            this.y = null;
            this.w = null;
            this.s = null;
            this.t = null;
            this.q = null;
            this.z = null;
            this.o = null;
            this.ab = null;
            this.u = null;
            this.v = null;
            this.x = null;
            this.r = null;
            this.ac = null;
            this.cu = null;
            this.cn = null;
            this.co = null;
            this.cv = null;
            this.cw = null;
            this.dp = null;
            this.dn = null;
            this.dz = null;
            this.d8 = null;
            this.d6 = null;
            this.d9 = null;
            this.d7 = null;
            this.d1 = null;
            this.d0 = null;
            this.d3 = null;
            this.d2 = null;
            this.d5 = null;
            this.d4 = null;
            this.dr = null;
            this.dq = null;
            this.i = null;
            this.g = null;
            this.n = null;
            this.f = null;
            this.h = null;
            this.l = null;
            this.m = null;
            this.k = null;
            this.j = null;
            this.c7 = null;
            this.cy = null;
            this.c2 = null;
            this.c1 = null;
            this.c3 = null;
            this.cx = null;
            this.c0 = null;
            this.cz = null;
            this.c5 = null;
            this.c4 = null;
            this.db = null;
            this.c8 = null;
            this.c9 = null;
            this.dc = null;
            this.df = null;
            this.c6 = null;
            this.dd = null;
            this.dl = null;
            this.da = null;
            this.dh = null;
            this.dj = null;
            this.dg = null;
            this.dk = null;
            this.di = null;
            this.dm = null;
            this.cs = null;
            this.cq = null;
            this.cr = null;
            this.cp = null;
            this.ae = null;
            this.ct = null;
            this.de = null;
        }
        get_type() {
            return "WebGridResourceStrings";
        }
        get type() {
            return this.get_type();
        }
        get igx_grid_groupByArea_message() {
            return this.cl;
        }
        set igx_grid_groupByArea_message(a) {
            this.cl = a;
            this.e("Igx_grid_groupByArea_message");
        }
        get igx_grid_groupByArea_select_message() {
            return this.cm;
        }
        set igx_grid_groupByArea_select_message(a) {
            this.cm = a;
            this.e("Igx_grid_groupByArea_select_message");
        }
        get igx_grid_groupByArea_deselect_message() {
            return this.ck;
        }
        set igx_grid_groupByArea_deselect_message(a) {
            this.ck = a;
            this.e("Igx_grid_groupByArea_deselect_message");
        }
        get igx_grid_emptyFilteredGrid_message() {
            return this.af;
        }
        set igx_grid_emptyFilteredGrid_message(a) {
            this.af = a;
            this.e("Igx_grid_emptyFilteredGrid_message");
        }
        get igx_grid_emptyGrid_message() {
            return this.ag;
        }
        set igx_grid_emptyGrid_message(a) {
            this.ag = a;
            this.e("Igx_grid_emptyGrid_message");
        }
        get igx_grid_filter() {
            return this.bf;
        }
        set igx_grid_filter(a) {
            this.bf = a;
            this.e("Igx_grid_filter");
        }
        get igx_grid_filter_row_close() {
            return this.b9;
        }
        set igx_grid_filter_row_close(a) {
            this.b9 = a;
            this.e("Igx_grid_filter_row_close");
        }
        get igx_grid_filter_row_reset() {
            return this.cc;
        }
        set igx_grid_filter_row_reset(a) {
            this.cc = a;
            this.e("Igx_grid_filter_row_reset");
        }
        get igx_grid_filter_row_placeholder() {
            return this.cb;
        }
        set igx_grid_filter_row_placeholder(a) {
            this.cb = a;
            this.e("Igx_grid_filter_row_placeholder");
        }
        get igx_grid_filter_row_boolean_placeholder() {
            return this.b8;
        }
        set igx_grid_filter_row_boolean_placeholder(a) {
            this.b8 = a;
            this.e("Igx_grid_filter_row_boolean_placeholder");
        }
        get igx_grid_filter_row_date_placeholder() {
            return this.ca;
        }
        set igx_grid_filter_row_date_placeholder(a) {
            this.ca = a;
            this.e("Igx_grid_filter_row_date_placeholder");
        }
        get igx_grid_filter_row_time_placeholder() {
            return this.cd;
        }
        set igx_grid_filter_row_time_placeholder(a) {
            this.cd = a;
            this.e("Igx_grid_filter_row_time_placeholder");
        }
        get igx_grid_filter_operator_and() {
            return this.b6;
        }
        set igx_grid_filter_operator_and(a) {
            this.b6 = a;
            this.e("Igx_grid_filter_operator_and");
        }
        get igx_grid_complex_filter() {
            return this.ad;
        }
        set igx_grid_complex_filter(a) {
            this.ad = a;
            this.e("Igx_grid_complex_filter");
        }
        get igx_grid_filter_operator_or() {
            return this.b7;
        }
        set igx_grid_filter_operator_or(a) {
            this.b7 = a;
            this.e("Igx_grid_filter_operator_or");
        }
        get igx_grid_filter_contains() {
            return this.bn;
        }
        set igx_grid_filter_contains(a) {
            this.bn = a;
            this.e("Igx_grid_filter_contains");
        }
        get igx_grid_filter_doesNotContain() {
            return this.bo;
        }
        set igx_grid_filter_doesNotContain(a) {
            this.bo = a;
            this.e("Igx_grid_filter_doesNotContain");
        }
        get igx_grid_filter_startsWith() {
            return this.ce;
        }
        set igx_grid_filter_startsWith(a) {
            this.ce = a;
            this.e("Igx_grid_filter_startsWith");
        }
        get igx_grid_filter_endsWith() {
            return this.br;
        }
        set igx_grid_filter_endsWith(a) {
            this.br = a;
            this.e("Igx_grid_filter_endsWith");
        }
        get igx_grid_filter_equals() {
            return this.bs;
        }
        set igx_grid_filter_equals(a) {
            this.bs = a;
            this.e("Igx_grid_filter_equals");
        }
        get igx_grid_filter_doesNotEqual() {
            return this.bp;
        }
        set igx_grid_filter_doesNotEqual(a) {
            this.bp = a;
            this.e("Igx_grid_filter_doesNotEqual");
        }
        get igx_grid_filter_empty() {
            return this.bq;
        }
        set igx_grid_filter_empty(a) {
            this.bq = a;
            this.e("Igx_grid_filter_empty");
        }
        get igx_grid_filter_notEmpty() {
            return this.b3;
        }
        set igx_grid_filter_notEmpty(a) {
            this.b3 = a;
            this.e("Igx_grid_filter_notEmpty");
        }
        get igx_grid_filter_null() {
            return this.b5;
        }
        set igx_grid_filter_null(a) {
            this.b5 = a;
            this.e("Igx_grid_filter_null");
        }
        get igx_grid_filter_notNull() {
            return this.b4;
        }
        set igx_grid_filter_notNull(a) {
            this.b4 = a;
            this.e("Igx_grid_filter_notNull");
        }
        get igx_grid_filter_before() {
            return this.bl;
        }
        set igx_grid_filter_before(a) {
            this.bl = a;
            this.e("Igx_grid_filter_before");
        }
        get igx_grid_filter_after() {
            return this.bg;
        }
        set igx_grid_filter_after(a) {
            this.bg = a;
            this.e("Igx_grid_filter_after");
        }
        get igx_grid_filter_at() {
            return this.bi;
        }
        set igx_grid_filter_at(a) {
            this.bi = a;
            this.e("Igx_grid_filter_at");
        }
        get igx_grid_filter_not_at() {
            return this.b2;
        }
        set igx_grid_filter_not_at(a) {
            this.b2 = a;
            this.e("Igx_grid_filter_not_at");
        }
        get igx_grid_filter_at_before() {
            return this.bk;
        }
        set igx_grid_filter_at_before(a) {
            this.bk = a;
            this.e("Igx_grid_filter_at_before");
        }
        get igx_grid_filter_at_after() {
            return this.bj;
        }
        set igx_grid_filter_at_after(a) {
            this.bj = a;
            this.e("Igx_grid_filter_at_after");
        }
        get igx_grid_filter_today() {
            return this.ch;
        }
        set igx_grid_filter_today(a) {
            this.ch = a;
            this.e("Igx_grid_filter_today");
        }
        get igx_grid_filter_yesterday() {
            return this.cj;
        }
        set igx_grid_filter_yesterday(a) {
            this.cj = a;
            this.e("Igx_grid_filter_yesterday");
        }
        get igx_grid_filter_thisMonth() {
            return this.cf;
        }
        set igx_grid_filter_thisMonth(a) {
            this.cf = a;
            this.e("Igx_grid_filter_thisMonth");
        }
        get igx_grid_filter_lastMonth() {
            return this.bw;
        }
        set igx_grid_filter_lastMonth(a) {
            this.bw = a;
            this.e("Igx_grid_filter_lastMonth");
        }
        get igx_grid_filter_nextMonth() {
            return this.b0;
        }
        set igx_grid_filter_nextMonth(a) {
            this.b0 = a;
            this.e("Igx_grid_filter_nextMonth");
        }
        get igx_grid_filter_thisYear() {
            return this.cg;
        }
        set igx_grid_filter_thisYear(a) {
            this.cg = a;
            this.e("Igx_grid_filter_thisYear");
        }
        get igx_grid_filter_lastYear() {
            return this.bx;
        }
        set igx_grid_filter_lastYear(a) {
            this.bx = a;
            this.e("Igx_grid_filter_lastYear");
        }
        get igx_grid_filter_nextYear() {
            return this.b1;
        }
        set igx_grid_filter_nextYear(a) {
            this.b1 = a;
            this.e("Igx_grid_filter_nextYear");
        }
        get igx_grid_filter_greaterThan() {
            return this.bu;
        }
        set igx_grid_filter_greaterThan(a) {
            this.bu = a;
            this.e("Igx_grid_filter_greaterThan");
        }
        get igx_grid_filter_lessThan() {
            return this.by;
        }
        set igx_grid_filter_lessThan(a) {
            this.by = a;
            this.e("Igx_grid_filter_lessThan");
        }
        get igx_grid_filter_greaterThanOrEqualTo() {
            return this.bv;
        }
        set igx_grid_filter_greaterThanOrEqualTo(a) {
            this.bv = a;
            this.e("Igx_grid_filter_greaterThanOrEqualTo");
        }
        get igx_grid_filter_lessThanOrEqualTo() {
            return this.bz;
        }
        set igx_grid_filter_lessThanOrEqualTo(a) {
            this.bz = a;
            this.e("Igx_grid_filter_lessThanOrEqualTo");
        }
        get igx_grid_filter_true() {
            return this.ci;
        }
        set igx_grid_filter_true(a) {
            this.ci = a;
            this.e("Igx_grid_filter_true");
        }
        get igx_grid_filter_false() {
            return this.bt;
        }
        set igx_grid_filter_false(a) {
            this.bt = a;
            this.e("Igx_grid_filter_false");
        }
        get igx_grid_filter_all() {
            return this.bh;
        }
        set igx_grid_filter_all(a) {
            this.bh = a;
            this.e("Igx_grid_filter_all");
        }
        get igx_grid_filter_condition_placeholder() {
            return this.bm;
        }
        set igx_grid_filter_condition_placeholder(a) {
            this.bm = a;
            this.e("Igx_grid_filter_condition_placeholder");
        }
        get igx_grid_summary_count() {
            return this.dt;
        }
        set igx_grid_summary_count(a) {
            this.dt = a;
            this.e("Igx_grid_summary_count");
        }
        get igx_grid_summary_min() {
            return this.dx;
        }
        set igx_grid_summary_min(a) {
            this.dx = a;
            this.e("Igx_grid_summary_min");
        }
        get igx_grid_summary_max() {
            return this.dw;
        }
        set igx_grid_summary_max(a) {
            this.dw = a;
            this.e("Igx_grid_summary_max");
        }
        get igx_grid_summary_sum() {
            return this.dy;
        }
        set igx_grid_summary_sum(a) {
            this.dy = a;
            this.e("Igx_grid_summary_sum");
        }
        get igx_grid_summary_average() {
            return this.ds;
        }
        set igx_grid_summary_average(a) {
            this.ds = a;
            this.e("Igx_grid_summary_average");
        }
        get igx_grid_summary_earliest() {
            return this.du;
        }
        set igx_grid_summary_earliest(a) {
            this.du = a;
            this.e("Igx_grid_summary_earliest");
        }
        get igx_grid_summary_latest() {
            return this.dv;
        }
        set igx_grid_summary_latest(a) {
            this.dv = a;
            this.e("Igx_grid_summary_latest");
        }
        get igx_grid_excel_filter_moving_left() {
            return this.av;
        }
        set igx_grid_excel_filter_moving_left(a) {
            this.av = a;
            this.e("Igx_grid_excel_filter_moving_left");
        }
        get igx_grid_excel_filter_moving_left_short() {
            return this.aw;
        }
        set igx_grid_excel_filter_moving_left_short(a) {
            this.aw = a;
            this.e("Igx_grid_excel_filter_moving_left_short");
        }
        get igx_grid_excel_filter_moving_right() {
            return this.ax;
        }
        set igx_grid_excel_filter_moving_right(a) {
            this.ax = a;
            this.e("Igx_grid_excel_filter_moving_right");
        }
        get igx_grid_excel_filter_moving_right_short() {
            return this.ay;
        }
        set igx_grid_excel_filter_moving_right_short(a) {
            this.ay = a;
            this.e("Igx_grid_excel_filter_moving_right_short");
        }
        get igx_grid_excel_filter_moving_header() {
            return this.au;
        }
        set igx_grid_excel_filter_moving_header(a) {
            this.au = a;
            this.e("Igx_grid_excel_filter_moving_header");
        }
        get igx_grid_excel_filter_sorting_asc() {
            return this.az;
        }
        set igx_grid_excel_filter_sorting_asc(a) {
            this.az = a;
            this.e("Igx_grid_excel_filter_sorting_asc");
        }
        get igx_grid_excel_filter_sorting_asc_short() {
            return this.a0;
        }
        set igx_grid_excel_filter_sorting_asc_short(a) {
            this.a0 = a;
            this.e("Igx_grid_excel_filter_sorting_asc_short");
        }
        get igx_grid_excel_filter_sorting_desc() {
            return this.a1;
        }
        set igx_grid_excel_filter_sorting_desc(a) {
            this.a1 = a;
            this.e("Igx_grid_excel_filter_sorting_desc");
        }
        get igx_grid_excel_filter_sorting_desc_short() {
            return this.a2;
        }
        set igx_grid_excel_filter_sorting_desc_short(a) {
            this.a2 = a;
            this.e("Igx_grid_excel_filter_sorting_desc_short");
        }
        get igx_grid_excel_filter_sorting_header() {
            return this.a3;
        }
        set igx_grid_excel_filter_sorting_header(a) {
            this.a3 = a;
            this.e("Igx_grid_excel_filter_sorting_header");
        }
        get igx_grid_excel_filter_clear() {
            return this.at;
        }
        set igx_grid_excel_filter_clear(a) {
            this.at = a;
            this.e("Igx_grid_excel_filter_clear");
        }
        get igx_grid_excel_custom_dialog_add() {
            return this.an;
        }
        set igx_grid_excel_custom_dialog_add(a) {
            this.an = a;
            this.e("Igx_grid_excel_custom_dialog_add");
        }
        get igx_grid_excel_custom_dialog_clear() {
            return this.ao;
        }
        set igx_grid_excel_custom_dialog_clear(a) {
            this.ao = a;
            this.e("Igx_grid_excel_custom_dialog_clear");
        }
        get igx_grid_excel_custom_dialog_header() {
            return this.ap;
        }
        set igx_grid_excel_custom_dialog_header(a) {
            this.ap = a;
            this.e("Igx_grid_excel_custom_dialog_header");
        }
        get igx_grid_excel_cancel() {
            return this.al;
        }
        set igx_grid_excel_cancel(a) {
            this.al = a;
            this.e("Igx_grid_excel_cancel");
        }
        get igx_grid_excel_apply() {
            return this.ai;
        }
        set igx_grid_excel_apply(a) {
            this.ai = a;
            this.e("Igx_grid_excel_apply");
        }
        get igx_grid_excel_search_placeholder() {
            return this.a8;
        }
        set igx_grid_excel_search_placeholder(a) {
            this.a8 = a;
            this.e("Igx_grid_excel_search_placeholder");
        }
        get igx_grid_excel_select_all() {
            return this.ba;
        }
        set igx_grid_excel_select_all(a) {
            this.ba = a;
            this.e("Igx_grid_excel_select_all");
        }
        get igx_grid_excel_select_all_search_results() {
            return this.bb;
        }
        set igx_grid_excel_select_all_search_results(a) {
            this.bb = a;
            this.e("Igx_grid_excel_select_all_search_results");
        }
        get igx_grid_excel_add_to_filter() {
            return this.ah;
        }
        set igx_grid_excel_add_to_filter(a) {
            this.ah = a;
            this.e("Igx_grid_excel_add_to_filter");
        }
        get igx_grid_excel_blanks() {
            return this.aj;
        }
        set igx_grid_excel_blanks(a) {
            this.aj = a;
            this.e("Igx_grid_excel_blanks");
        }
        get igx_grid_excel_hide() {
            return this.a4;
        }
        set igx_grid_excel_hide(a) {
            this.a4 = a;
            this.e("Igx_grid_excel_hide");
        }
        get igx_grid_excel_show() {
            return this.bc;
        }
        set igx_grid_excel_show(a) {
            this.bc = a;
            this.e("Igx_grid_excel_show");
        }
        get igx_grid_excel_pin() {
            return this.a7;
        }
        set igx_grid_excel_pin(a) {
            this.a7 = a;
            this.e("Igx_grid_excel_pin");
        }
        get igx_grid_excel_unpin() {
            return this.be;
        }
        set igx_grid_excel_unpin(a) {
            this.be = a;
            this.e("Igx_grid_excel_unpin");
        }
        get igx_grid_excel_select() {
            return this.a9;
        }
        set igx_grid_excel_select(a) {
            this.a9 = a;
            this.e("Igx_grid_excel_select");
        }
        get igx_grid_excel_deselect() {
            return this.as;
        }
        set igx_grid_excel_deselect(a) {
            this.as = a;
            this.e("Igx_grid_excel_deselect");
        }
        get igx_grid_excel_text_filter() {
            return this.bd;
        }
        set igx_grid_excel_text_filter(a) {
            this.bd = a;
            this.e("Igx_grid_excel_text_filter");
        }
        get igx_grid_excel_number_filter() {
            return this.a6;
        }
        set igx_grid_excel_number_filter(a) {
            this.a6 = a;
            this.e("Igx_grid_excel_number_filter");
        }
        get igx_grid_excel_date_filter() {
            return this.ar;
        }
        set igx_grid_excel_date_filter(a) {
            this.ar = a;
            this.e("Igx_grid_excel_date_filter");
        }
        get igx_grid_excel_boolean_filter() {
            return this.ak;
        }
        set igx_grid_excel_boolean_filter(a) {
            this.ak = a;
            this.e("Igx_grid_excel_boolean_filter");
        }
        get igx_grid_excel_currency_filter() {
            return this.am;
        }
        set igx_grid_excel_currency_filter(a) {
            this.am = a;
            this.e("Igx_grid_excel_currency_filter");
        }
        get igx_grid_excel_custom_filter() {
            return this.aq;
        }
        set igx_grid_excel_custom_filter(a) {
            this.aq = a;
            this.e("Igx_grid_excel_custom_filter");
        }
        get igx_grid_excel_no_matches() {
            return this.a5;
        }
        set igx_grid_excel_no_matches(a) {
            this.a5 = a;
            this.e("Igx_grid_excel_no_matches");
        }
        get igx_grid_advanced_filter_title() {
            return this.aa;
        }
        set igx_grid_advanced_filter_title(a) {
            this.aa = a;
            this.e("Igx_grid_advanced_filter_title");
        }
        get igx_grid_advanced_filter_and_group() {
            return this.p;
        }
        set igx_grid_advanced_filter_and_group(a) {
            this.p = a;
            this.e("Igx_grid_advanced_filter_and_group");
        }
        get igx_grid_advanced_filter_or_group() {
            return this.y;
        }
        set igx_grid_advanced_filter_or_group(a) {
            this.y = a;
            this.e("Igx_grid_advanced_filter_or_group");
        }
        get igx_grid_advanced_filter_end_group() {
            return this.w;
        }
        set igx_grid_advanced_filter_end_group(a) {
            this.w = a;
            this.e("Igx_grid_advanced_filter_end_group");
        }
        get igx_grid_advanced_filter_create_and_group() {
            return this.s;
        }
        set igx_grid_advanced_filter_create_and_group(a) {
            this.s = a;
            this.e("Igx_grid_advanced_filter_create_and_group");
        }
        get igx_grid_advanced_filter_create_or_group() {
            return this.t;
        }
        set igx_grid_advanced_filter_create_or_group(a) {
            this.t = a;
            this.e("Igx_grid_advanced_filter_create_or_group");
        }
        get igx_grid_advanced_filter_and_label() {
            return this.q;
        }
        set igx_grid_advanced_filter_and_label(a) {
            this.q = a;
            this.e("Igx_grid_advanced_filter_and_label");
        }
        get igx_grid_advanced_filter_or_label() {
            return this.z;
        }
        set igx_grid_advanced_filter_or_label(a) {
            this.z = a;
            this.e("Igx_grid_advanced_filter_or_label");
        }
        get igx_grid_advanced_filter_add_condition() {
            return this.o;
        }
        set igx_grid_advanced_filter_add_condition(a) {
            this.o = a;
            this.e("Igx_grid_advanced_filter_add_condition");
        }
        get igx_grid_advanced_filter_ungroup() {
            return this.ab;
        }
        set igx_grid_advanced_filter_ungroup(a) {
            this.ab = a;
            this.e("Igx_grid_advanced_filter_ungroup");
        }
        get igx_grid_advanced_filter_delete() {
            return this.u;
        }
        set igx_grid_advanced_filter_delete(a) {
            this.u = a;
            this.e("Igx_grid_advanced_filter_delete");
        }
        get igx_grid_advanced_filter_delete_filters() {
            return this.v;
        }
        set igx_grid_advanced_filter_delete_filters(a) {
            this.v = a;
            this.e("Igx_grid_advanced_filter_delete_filters");
        }
        get igx_grid_advanced_filter_initial_text() {
            return this.x;
        }
        set igx_grid_advanced_filter_initial_text(a) {
            this.x = a;
            this.e("Igx_grid_advanced_filter_initial_text");
        }
        get igx_grid_advanced_filter_column_placeholder() {
            return this.r;
        }
        set igx_grid_advanced_filter_column_placeholder(a) {
            this.r = a;
            this.e("Igx_grid_advanced_filter_column_placeholder");
        }
        get igx_grid_advanced_filter_value_placeholder() {
            return this.ac;
        }
        set igx_grid_advanced_filter_value_placeholder(a) {
            this.ac = a;
            this.e("Igx_grid_advanced_filter_value_placeholder");
        }
        get igx_grid_pinned_row_indicator() {
            return this.cu;
        }
        set igx_grid_pinned_row_indicator(a) {
            this.cu = a;
            this.e("Igx_grid_pinned_row_indicator");
        }
        get igx_grid_hiding_check_all_label() {
            return this.cn;
        }
        set igx_grid_hiding_check_all_label(a) {
            this.cn = a;
            this.e("Igx_grid_hiding_check_all_label");
        }
        get igx_grid_hiding_uncheck_all_label() {
            return this.co;
        }
        set igx_grid_hiding_uncheck_all_label(a) {
            this.co = a;
            this.e("Igx_grid_hiding_uncheck_all_label");
        }
        get igx_grid_pinning_check_all_label() {
            return this.cv;
        }
        set igx_grid_pinning_check_all_label(a) {
            this.cv = a;
            this.e("Igx_grid_pinning_check_all_label");
        }
        get igx_grid_pinning_uncheck_all_label() {
            return this.cw;
        }
        set igx_grid_pinning_uncheck_all_label(a) {
            this.cw = a;
            this.e("Igx_grid_pinning_uncheck_all_label");
        }
        get igx_grid_row_edit_btn_done() {
            return this.dp;
        }
        set igx_grid_row_edit_btn_done(a) {
            this.dp = a;
            this.e("Igx_grid_row_edit_btn_done");
        }
        get igx_grid_row_edit_btn_cancel() {
            return this.dn;
        }
        set igx_grid_row_edit_btn_cancel(a) {
            this.dn = a;
            this.e("Igx_grid_row_edit_btn_cancel");
        }
        get igx_grid_toolbar_actions_filter_prompt() {
            return this.dz;
        }
        set igx_grid_toolbar_actions_filter_prompt(a) {
            this.dz = a;
            this.e("Igx_grid_toolbar_actions_filter_prompt");
        }
        get igx_grid_toolbar_pinning_button_tooltip() {
            return this.d8;
        }
        set igx_grid_toolbar_pinning_button_tooltip(a) {
            this.d8 = a;
            this.e("Igx_grid_toolbar_pinning_button_tooltip");
        }
        get igx_grid_toolbar_hiding_button_tooltip() {
            return this.d6;
        }
        set igx_grid_toolbar_hiding_button_tooltip(a) {
            this.d6 = a;
            this.e("Igx_grid_toolbar_hiding_button_tooltip");
        }
        get igx_grid_toolbar_pinning_title() {
            return this.d9;
        }
        set igx_grid_toolbar_pinning_title(a) {
            this.d9 = a;
            this.e("Igx_grid_toolbar_pinning_title");
        }
        get igx_grid_toolbar_hiding_title() {
            return this.d7;
        }
        set igx_grid_toolbar_hiding_title(a) {
            this.d7 = a;
            this.e("Igx_grid_toolbar_hiding_title");
        }
        get igx_grid_toolbar_advanced_filtering_button_tooltip() {
            return this.d1;
        }
        set igx_grid_toolbar_advanced_filtering_button_tooltip(a) {
            this.d1 = a;
            this.e("Igx_grid_toolbar_advanced_filtering_button_tooltip");
        }
        get igx_grid_toolbar_advanced_filtering_button_label() {
            return this.d0;
        }
        set igx_grid_toolbar_advanced_filtering_button_label(a) {
            this.d0 = a;
            this.e("Igx_grid_toolbar_advanced_filtering_button_label");
        }
        get igx_grid_toolbar_exporter_button_tooltip() {
            return this.d3;
        }
        set igx_grid_toolbar_exporter_button_tooltip(a) {
            this.d3 = a;
            this.e("Igx_grid_toolbar_exporter_button_tooltip");
        }
        get igx_grid_toolbar_exporter_button_label() {
            return this.d2;
        }
        set igx_grid_toolbar_exporter_button_label(a) {
            this.d2 = a;
            this.e("Igx_grid_toolbar_exporter_button_label");
        }
        get igx_grid_toolbar_exporter_excel_entry_text() {
            return this.d5;
        }
        set igx_grid_toolbar_exporter_excel_entry_text(a) {
            this.d5 = a;
            this.e("Igx_grid_toolbar_exporter_excel_entry_text");
        }
        get igx_grid_toolbar_exporter_csv_entry_text() {
            return this.d4;
        }
        set igx_grid_toolbar_exporter_csv_entry_text(a) {
            this.d4 = a;
            this.e("Igx_grid_toolbar_exporter_csv_entry_text");
        }
        get igx_grid_snackbar_addrow_label() {
            return this.dr;
        }
        set igx_grid_snackbar_addrow_label(a) {
            this.dr = a;
            this.e("Igx_grid_snackbar_addrow_label");
        }
        get igx_grid_snackbar_addrow_actiontext() {
            return this.dq;
        }
        set igx_grid_snackbar_addrow_actiontext(a) {
            this.dq = a;
            this.e("Igx_grid_snackbar_addrow_actiontext");
        }
        get igx_grid_actions_edit_label() {
            return this.i;
        }
        set igx_grid_actions_edit_label(a) {
            this.i = a;
            this.e("Igx_grid_actions_edit_label");
        }
        get igx_grid_actions_add_label() {
            return this.g;
        }
        set igx_grid_actions_add_label(a) {
            this.g = a;
            this.e("Igx_grid_actions_add_label");
        }
        get igx_grid_add_row_label() {
            return this.n;
        }
        set igx_grid_add_row_label(a) {
            this.n = a;
            this.e("Igx_grid_add_row_label");
        }
        get igx_grid_actions_add_child_label() {
            return this.f;
        }
        set igx_grid_actions_add_child_label(a) {
            this.f = a;
            this.e("Igx_grid_actions_add_child_label");
        }
        get igx_grid_actions_delete_label() {
            return this.h;
        }
        set igx_grid_actions_delete_label(a) {
            this.h = a;
            this.e("Igx_grid_actions_delete_label");
        }
        get igx_grid_actions_pin_label() {
            return this.l;
        }
        set igx_grid_actions_pin_label(a) {
            this.l = a;
            this.e("Igx_grid_actions_pin_label");
        }
        get igx_grid_actions_unpin_label() {
            return this.m;
        }
        set igx_grid_actions_unpin_label(a) {
            this.m = a;
            this.e("Igx_grid_actions_unpin_label");
        }
        get igx_grid_actions_jumpUp_label() {
            return this.k;
        }
        set igx_grid_actions_jumpUp_label(a) {
            this.k = a;
            this.e("Igx_grid_actions_jumpUp_label");
        }
        get igx_grid_actions_jumpDown_label() {
            return this.j;
        }
        set igx_grid_actions_jumpDown_label(a) {
            this.j = a;
            this.e("Igx_grid_actions_jumpDown_label");
        }
        get igx_grid_pivot_date_dimension_total() {
            return this.c7;
        }
        set igx_grid_pivot_date_dimension_total(a) {
            this.c7 = a;
            this.e("Igx_grid_pivot_date_dimension_total");
        }
        get igx_grid_pivot_aggregate_count() {
            return this.cy;
        }
        set igx_grid_pivot_aggregate_count(a) {
            this.cy = a;
            this.e("Igx_grid_pivot_aggregate_count");
        }
        get igx_grid_pivot_aggregate_min() {
            return this.c2;
        }
        set igx_grid_pivot_aggregate_min(a) {
            this.c2 = a;
            this.e("Igx_grid_pivot_aggregate_min");
        }
        get igx_grid_pivot_aggregate_max() {
            return this.c1;
        }
        set igx_grid_pivot_aggregate_max(a) {
            this.c1 = a;
            this.e("Igx_grid_pivot_aggregate_max");
        }
        get igx_grid_pivot_aggregate_sum() {
            return this.c3;
        }
        set igx_grid_pivot_aggregate_sum(a) {
            this.c3 = a;
            this.e("Igx_grid_pivot_aggregate_sum");
        }
        get igx_grid_pivot_aggregate_avg() {
            return this.cx;
        }
        set igx_grid_pivot_aggregate_avg(a) {
            this.cx = a;
            this.e("Igx_grid_pivot_aggregate_avg");
        }
        get igx_grid_pivot_aggregate_date_latest() {
            return this.c0;
        }
        set igx_grid_pivot_aggregate_date_latest(a) {
            this.c0 = a;
            this.e("Igx_grid_pivot_aggregate_date_latest");
        }
        get igx_grid_pivot_aggregate_date_earliest() {
            return this.cz;
        }
        set igx_grid_pivot_aggregate_date_earliest(a) {
            this.cz = a;
            this.e("Igx_grid_pivot_aggregate_date_earliest");
        }
        get igx_grid_pivot_aggregate_time_latest() {
            return this.c5;
        }
        set igx_grid_pivot_aggregate_time_latest(a) {
            this.c5 = a;
            this.e("Igx_grid_pivot_aggregate_time_latest");
        }
        get igx_grid_pivot_aggregate_time_earliest() {
            return this.c4;
        }
        set igx_grid_pivot_aggregate_time_earliest(a) {
            this.c4 = a;
            this.e("Igx_grid_pivot_aggregate_time_earliest");
        }
        get igx_grid_pivot_empty_row_drop_area() {
            return this.db;
        }
        set igx_grid_pivot_empty_row_drop_area(a) {
            this.db = a;
            this.e("Igx_grid_pivot_empty_row_drop_area");
        }
        get igx_grid_pivot_empty_column_drop_area() {
            return this.c8;
        }
        set igx_grid_pivot_empty_column_drop_area(a) {
            this.c8 = a;
            this.e("Igx_grid_pivot_empty_column_drop_area");
        }
        get igx_grid_pivot_empty_filter_drop_area() {
            return this.c9;
        }
        set igx_grid_pivot_empty_filter_drop_area(a) {
            this.c9 = a;
            this.e("Igx_grid_pivot_empty_filter_drop_area");
        }
        get igx_grid_pivot_empty_value_drop_area() {
            return this.dc;
        }
        set igx_grid_pivot_empty_value_drop_area(a) {
            this.dc = a;
            this.e("Igx_grid_pivot_empty_value_drop_area");
        }
        get igx_grid_pivot_row_drop_chip() {
            return this.df;
        }
        set igx_grid_pivot_row_drop_chip(a) {
            this.df = a;
            this.e("Igx_grid_pivot_row_drop_chip");
        }
        get igx_grid_pivot_column_drop_chip() {
            return this.c6;
        }
        set igx_grid_pivot_column_drop_chip(a) {
            this.c6 = a;
            this.e("Igx_grid_pivot_column_drop_chip");
        }
        get igx_grid_pivot_filter_drop_chip() {
            return this.dd;
        }
        set igx_grid_pivot_filter_drop_chip(a) {
            this.dd = a;
            this.e("Igx_grid_pivot_filter_drop_chip");
        }
        get igx_grid_pivot_value_drop_chip() {
            return this.dl;
        }
        set igx_grid_pivot_value_drop_chip(a) {
            this.dl = a;
            this.e("Igx_grid_pivot_value_drop_chip");
        }
        get igx_grid_pivot_empty_message() {
            return this.da;
        }
        set igx_grid_pivot_empty_message(a) {
            this.da = a;
            this.e("Igx_grid_pivot_empty_message");
        }
        get igx_grid_pivot_selector_filters() {
            return this.dh;
        }
        set igx_grid_pivot_selector_filters(a) {
            this.dh = a;
            this.e("Igx_grid_pivot_selector_filters");
        }
        get igx_grid_pivot_selector_rows() {
            return this.dj;
        }
        set igx_grid_pivot_selector_rows(a) {
            this.dj = a;
            this.e("Igx_grid_pivot_selector_rows");
        }
        get igx_grid_pivot_selector_columns() {
            return this.dg;
        }
        set igx_grid_pivot_selector_columns(a) {
            this.dg = a;
            this.e("Igx_grid_pivot_selector_columns");
        }
        get igx_grid_pivot_selector_values() {
            return this.dk;
        }
        set igx_grid_pivot_selector_values(a) {
            this.dk = a;
            this.e("Igx_grid_pivot_selector_values");
        }
        get igx_grid_pivot_selector_panel_empty() {
            return this.di;
        }
        set igx_grid_pivot_selector_panel_empty(a) {
            this.di = a;
            this.e("Igx_grid_pivot_selector_panel_empty");
        }
        get igx_grid_required_validation_error() {
            return this.dm;
        }
        set igx_grid_required_validation_error(a) {
            this.dm = a;
            this.e("Igx_grid_required_validation_error");
        }
        get igx_grid_min_validation_error() {
            return this.cs;
        }
        set igx_grid_min_validation_error(a) {
            this.cs = a;
            this.e("Igx_grid_min_validation_error");
        }
        get igx_grid_max_validation_error() {
            return this.cq;
        }
        set igx_grid_max_validation_error(a) {
            this.cq = a;
            this.e("Igx_grid_max_validation_error");
        }
        get igx_grid_min_length_validation_error() {
            return this.cr;
        }
        set igx_grid_min_length_validation_error(a) {
            this.cr = a;
            this.e("Igx_grid_min_length_validation_error");
        }
        get igx_grid_max_length_validation_error() {
            return this.cp;
        }
        set igx_grid_max_length_validation_error(a) {
            this.cp = a;
            this.e("Igx_grid_max_length_validation_error");
        }
        get igx_grid_email_validation_error() {
            return this.ae;
        }
        set igx_grid_email_validation_error(a) {
            this.ae = a;
            this.e("Igx_grid_email_validation_error");
        }
        get igx_grid_pattern_validation_error() {
            return this.ct;
        }
        set igx_grid_pattern_validation_error(a) {
            this.ct = a;
            this.e("Igx_grid_pattern_validation_error");
        }
        get igx_grid_pivot_no_aggregator() {
            return this.de;
        }
        set igx_grid_pivot_no_aggregator(a) {
            this.de = a;
            this.e("Igx_grid_pivot_no_aggregator");
        }
    }
    WebGridResourceStringsDescription.$t = /*@__PURE__*/ markType(WebGridResourceStringsDescription, 'WebGridResourceStringsDescription', Description.$);
    return WebGridResourceStringsDescription;
})();
