// igniteui-react-core/esm2015/lib/WebGroupingDoneEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGroupingDoneEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebGroupingDoneEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.k = null;
            this.l = null;
            this.m = null;
        }
        get_type() {
            return "WebGroupingDoneEventArgsDetail";
        }
        get expressions() {
            return this.k;
        }
        set expressions(a) {
            this.k = a;
            this.e("Expressions");
        }
        get groupedColumns() {
            return this.l;
        }
        set groupedColumns(a) {
            this.l = a;
            this.e("GroupedColumns");
        }
        get ungroupedColumns() {
            return this.m;
        }
        set ungroupedColumns(a) {
            this.m = a;
            this.e("UngroupedColumns");
        }
    }
    WebGroupingDoneEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebGroupingDoneEventArgsDetailDescription, 'WebGroupingDoneEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebGroupingDoneEventArgsDetailDescription.__marshalByValue1 = true;
    WebGroupingDoneEventArgsDetailDescription.__marshalByValueAlias1 = "GroupingDoneEventArgsDetail";
    return WebGroupingDoneEventArgsDetailDescription;
})();
