// igniteui-react-core/esm2015/lib/WebGroupByKeyDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGroupByKeyDescription = /*@__PURE__*/ (() => {
    class WebGroupByKeyDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = null;
        }
        get_type() {
            return "WebGroupByKey";
        }
        get type() {
            return this.get_type();
        }
        get fieldName() {
            return this.h;
        }
        set fieldName(a) {
            this.h = a;
            this.e("FieldName");
        }
        get value() {
            return this.f;
        }
        set value(a) {
            this.f = a;
            this.e("Value");
        }
    }
    WebGroupByKeyDescription.$t = /*@__PURE__*/ markType(WebGroupByKeyDescription, 'WebGroupByKeyDescription', Description.$);
    return WebGroupByKeyDescription;
})();
