// igniteui-react-core/esm2015/lib/WebGridClipboardEventDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGridClipboardEventDescription = /*@__PURE__*/ (() => {
    class WebGridClipboardEventDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = false;
        }
        get_type() {
            return "WebGridClipboardEvent";
        }
        get type() {
            return this.get_type();
        }
        get dataRef() {
            return this.h;
        }
        set dataRef(a) {
            this.h = a;
            this.e("DataRef");
        }
        get cancel() {
            return this.f;
        }
        set cancel(a) {
            this.f = a;
            this.e("Cancel");
        }
    }
    WebGridClipboardEventDescription.$t = /*@__PURE__*/ markType(WebGridClipboardEventDescription, 'WebGridClipboardEventDescription', Description.$);
    return WebGridClipboardEventDescription;
})();
