// igniteui-react-core/esm2015/lib/WebGridClipboardEventDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebGridClipboardEventDetailDescription = /*@__PURE__*/ (() => {
    class WebGridClipboardEventDetailDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.g = false;
        }
        get_type() {
            return "WebGridClipboardEventDetail";
        }
        get type() {
            return this.get_type();
        }
        get dataRef() {
            return this.j;
        }
        set dataRef(a) {
            this.j = a;
            this.e("DataRef");
        }
        get cancel() {
            return this.g;
        }
        set cancel(a) {
            this.g = a;
            this.e("Cancel");
        }
    }
    WebGridClipboardEventDetailDescription.$t = /*@__PURE__*/ markType(WebGridClipboardEventDetailDescription, 'WebGridClipboardEventDetailDescription', Description.$);
    WebGridClipboardEventDetailDescription.__marshalByValue = true;
    WebGridClipboardEventDetailDescription.__marshalByValueAlias = "GridClipboardEventDetail";
    return WebGridClipboardEventDetailDescription;
})();
