// igniteui-react-core/esm2015/lib/WebPivotValueDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotValueDescription = /*@__PURE__*/ (() => {
    class WebPivotValueDescription extends Description {
        constructor() {
            super();
            this.q = null;
            this.o = null;
            this.h = null;
            this.f = null;
            this.j = false;
            this.l = null;
            this.n = null;
            this.p = null;
        }
        get_type() {
            return "WebPivotValue";
        }
        get type() {
            return this.get_type();
        }
        get member() {
            return this.q;
        }
        set member(a) {
            this.q = a;
            this.e("Member");
        }
        get displayName() {
            return this.o;
        }
        set displayName(a) {
            this.o = a;
            this.e("DisplayName");
        }
        get aggregate() {
            return this.h;
        }
        set aggregate(a) {
            this.h = a;
            this.e("Aggregate");
        }
        get aggregateList() {
            return this.f;
        }
        set aggregateList(a) {
            this.f = a;
            this.e("AggregateList");
        }
        get enabled() {
            return this.j;
        }
        set enabled(a) {
            this.j = a;
            this.e("Enabled");
        }
        get styles() {
            return this.l;
        }
        set styles(a) {
            this.l = a;
            this.e("Styles");
        }
        get dataType() {
            return this.n;
        }
        set dataType(a) {
            this.n = a;
            this.e("DataType");
        }
        get formatterRef() {
            return this.p;
        }
        set formatterRef(a) {
            this.p = a;
            this.e("FormatterRef");
        }
    }
    WebPivotValueDescription.$t = /*@__PURE__*/ markType(WebPivotValueDescription, 'WebPivotValueDescription', Description.$);
    return WebPivotValueDescription;
})();
