// igniteui-react-core/esm2015/lib/WebPivotConfigurationDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotConfigurationDescription = /*@__PURE__*/ (() => {
    class WebPivotConfigurationDescription extends Description {
        constructor() {
            super();
            this.o = null;
            this.n = null;
            this.h = null;
            this.f = null;
            this.i = null;
            this.g = null;
            this.r = null;
        }
        get_type() {
            return "WebPivotConfiguration";
        }
        get type() {
            return this.get_type();
        }
        get rowStrategy() {
            return this.o;
        }
        set rowStrategy(a) {
            this.o = a;
            this.e("RowStrategy");
        }
        get columnStrategy() {
            return this.n;
        }
        set columnStrategy(a) {
            this.n = a;
            this.e("ColumnStrategy");
        }
        get rows() {
            return this.h;
        }
        set rows(a) {
            this.h = a;
            this.e("Rows");
        }
        get columns() {
            return this.f;
        }
        set columns(a) {
            this.f = a;
            this.e("Columns");
        }
        get values() {
            return this.i;
        }
        set values(a) {
            this.i = a;
            this.e("Values");
        }
        get filters() {
            return this.g;
        }
        set filters(a) {
            this.g = a;
            this.e("Filters");
        }
        get pivotKeys() {
            return this.r;
        }
        set pivotKeys(a) {
            this.r = a;
            this.e("PivotKeys");
        }
    }
    WebPivotConfigurationDescription.$t = /*@__PURE__*/ markType(WebPivotConfigurationDescription, 'WebPivotConfigurationDescription', Description.$);
    return WebPivotConfigurationDescription;
})();
