// igniteui-react-core/esm2015/lib/WebPivotAggregatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotAggregatorDescription = /*@__PURE__*/ (() => {
    class WebPivotAggregatorDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.i = null;
            this.g = null;
            this.f = null;
        }
        get_type() {
            return "WebPivotAggregator";
        }
        get type() {
            return this.get_type();
        }
        get key() {
            return this.h;
        }
        set key(a) {
            this.h = a;
            this.e("Key");
        }
        get label() {
            return this.i;
        }
        set label(a) {
            this.i = a;
            this.e("Label");
        }
        get aggregatorName() {
            return this.g;
        }
        set aggregatorName(a) {
            this.g = a;
            this.e("AggregatorName");
        }
        get aggregatorRef() {
            return this.f;
        }
        set aggregatorRef(a) {
            this.f = a;
            this.e("AggregatorRef");
        }
    }
    WebPivotAggregatorDescription.$t = /*@__PURE__*/ markType(WebPivotAggregatorDescription, 'WebPivotAggregatorDescription', Description.$);
    return WebPivotAggregatorDescription;
})();
