// igniteui-react-core/esm2015/lib/WebPivotKeysDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotKeysDescription = /*@__PURE__*/ (() => {
    class WebPivotKeysDescription extends Description {
        constructor() {
            super();
            this.g = null;
            this.j = null;
            this.f = null;
            this.i = null;
            this.h = null;
            this.k = null;
        }
        get_type() {
            return "WebPivotKeys";
        }
        get type() {
            return this.get_type();
        }
        get children() {
            return this.g;
        }
        set children(a) {
            this.g = a;
            this.e("Children");
        }
        get records() {
            return this.j;
        }
        set records(a) {
            this.j = a;
            this.e("Records");
        }
        get aggregations() {
            return this.f;
        }
        set aggregations(a) {
            this.f = a;
            this.e("Aggregations");
        }
        get level() {
            return this.i;
        }
        set level(a) {
            this.i = a;
            this.e("Level");
        }
        get columnDimensionSeparator() {
            return this.h;
        }
        set columnDimensionSeparator(a) {
            this.h = a;
            this.e("ColumnDimensionSeparator");
        }
        get rowDimensionSeparator() {
            return this.k;
        }
        set rowDimensionSeparator(a) {
            this.k = a;
            this.e("RowDimensionSeparator");
        }
    }
    WebPivotKeysDescription.$t = /*@__PURE__*/ markType(WebPivotKeysDescription, 'WebPivotKeysDescription', Description.$);
    return WebPivotKeysDescription;
})();
