// igniteui-react-core/esm2015/lib/WebPinColumnEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPinColumnEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebPinColumnEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.k = null;
            this.p = 0;
            this.n = false;
        }
        get_type() {
            return "WebPinColumnEventArgsDetail";
        }
        get column() {
            return this.k;
        }
        set column(a) {
            this.k = a;
            this.e("Column");
        }
        get insertAtIndex() {
            return this.p;
        }
        set insertAtIndex(a) {
            this.p = a;
            this.e("InsertAtIndex");
        }
        get isPinned() {
            return this.n;
        }
        set isPinned(a) {
            this.n = a;
            this.e("IsPinned");
        }
    }
    WebPinColumnEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebPinColumnEventArgsDetailDescription, 'WebPinColumnEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebPinColumnEventArgsDetailDescription.__marshalByValue1 = true;
    WebPinColumnEventArgsDetailDescription.__marshalByValueAlias1 = "PinColumnEventArgsDetail";
    return WebPinColumnEventArgsDetailDescription;
})();
