// igniteui-react-core/esm2015/lib/WebPageEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPageEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebPageEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.m = 0;
            this.l = 0;
        }
        get_type() {
            return "WebPageEventArgsDetail";
        }
        get previous() {
            return this.m;
        }
        set previous(a) {
            this.m = a;
            this.e("Previous");
        }
        get current() {
            return this.l;
        }
        set current(a) {
            this.l = a;
            this.e("Current");
        }
    }
    WebPageEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebPageEventArgsDetailDescription, 'WebPageEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebPageEventArgsDetailDescription.__marshalByValue1 = true;
    WebPageEventArgsDetailDescription.__marshalByValueAlias1 = "PageEventArgsDetail";
    return WebPageEventArgsDetailDescription;
})();
