// igniteui-react-core/esm2015/lib/WebPivotGridColumnDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotGridColumnDescription = /*@__PURE__*/ (() => {
    class WebPivotGridColumnDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.f = null;
            this.h = null;
        }
        get_type() {
            return "WebPivotGridColumn";
        }
        get type() {
            return this.get_type();
        }
        get field() {
            return this.j;
        }
        set field(a) {
            this.j = a;
            this.e("Field");
        }
        get dimensions() {
            return this.f;
        }
        set dimensions(a) {
            this.f = a;
            this.e("Dimensions");
        }
        get value() {
            return this.h;
        }
        set value(a) {
            this.h = a;
            this.e("Value");
        }
    }
    WebPivotGridColumnDescription.$t = /*@__PURE__*/ markType(WebPivotGridColumnDescription, 'WebPivotGridColumnDescription', Description.$);
    return WebPivotGridColumnDescription;
})();
