// igniteui-react-core/esm2015/lib/WebPivotValueDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotValueDetailDescription = /*@__PURE__*/ (() => {
    class WebPivotValueDetailDescription extends Description {
        constructor() {
            super();
            this.s = null;
            this.q = null;
            this.h = null;
            this.f = null;
            this.k = false;
            this.m = null;
            this.p = null;
            this.r = null;
        }
        get_type() {
            return "WebPivotValueDetail";
        }
        get type() {
            return this.get_type();
        }
        get member() {
            return this.s;
        }
        set member(a) {
            this.s = a;
            this.e("Member");
        }
        get displayName() {
            return this.q;
        }
        set displayName(a) {
            this.q = a;
            this.e("DisplayName");
        }
        get aggregate() {
            return this.h;
        }
        set aggregate(a) {
            this.h = a;
            this.e("Aggregate");
        }
        get aggregateList() {
            return this.f;
        }
        set aggregateList(a) {
            this.f = a;
            this.e("AggregateList");
        }
        get enabled() {
            return this.k;
        }
        set enabled(a) {
            this.k = a;
            this.e("Enabled");
        }
        get styles() {
            return this.m;
        }
        set styles(a) {
            this.m = a;
            this.e("Styles");
        }
        get dataType() {
            return this.p;
        }
        set dataType(a) {
            this.p = a;
            this.e("DataType");
        }
        get formatterRef() {
            return this.r;
        }
        set formatterRef(a) {
            this.r = a;
            this.e("FormatterRef");
        }
    }
    WebPivotValueDetailDescription.$t = /*@__PURE__*/ markType(WebPivotValueDetailDescription, 'WebPivotValueDetailDescription', Description.$);
    WebPivotValueDetailDescription.__marshalByValue = true;
    WebPivotValueDetailDescription.__marshalByValueAlias = "PivotValueDetail";
    return WebPivotValueDetailDescription;
})();
