// igniteui-react-core/esm2015/lib/WebPivotGridRecordDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPivotGridRecordDescription = /*@__PURE__*/ (() => {
    class WebPivotGridRecordDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.h = 0;
            this.f = null;
        }
        get_type() {
            return "WebPivotGridRecord";
        }
        get type() {
            return this.get_type();
        }
        get recordsRef() {
            return this.j;
        }
        set recordsRef(a) {
            this.j = a;
            this.e("RecordsRef");
        }
        get level() {
            return this.h;
        }
        set level(a) {
            this.h = a;
            this.e("Level");
        }
        get dimensions() {
            return this.f;
        }
        set dimensions(a) {
            this.f = a;
            this.e("Dimensions");
        }
    }
    WebPivotGridRecordDescription.$t = /*@__PURE__*/ markType(WebPivotGridRecordDescription, 'WebPivotGridRecordDescription', Description.$);
    return WebPivotGridRecordDescription;
})();
