// igniteui-react-core/esm2015/lib/WebRowToggleEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebRowToggleEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebRowToggleEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.p = null;
            this.m = false;
            this.l = false;
        }
        get_type() {
            return "WebRowToggleEventArgsDetail";
        }
        get rowID() {
            return this.p;
        }
        set rowID(a) {
            this.p = a;
            this.e("RowID");
        }
        get expanded() {
            return this.m;
        }
        set expanded(a) {
            this.m = a;
            this.e("Expanded");
        }
        get cancel() {
            return this.l;
        }
        set cancel(a) {
            this.l = a;
            this.e("Cancel");
        }
    }
    WebRowToggleEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebRowToggleEventArgsDetailDescription, 'WebRowToggleEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebRowToggleEventArgsDetailDescription.__marshalByValue1 = true;
    WebRowToggleEventArgsDetailDescription.__marshalByValueAlias1 = "RowToggleEventArgsDetail";
    return WebRowToggleEventArgsDetailDescription;
})();
