// igniteui-react-core/esm2015/lib/WebRowDragEndEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebRowDragEndEventArgsDetailDescription = /*@__PURE__*/ (() => {
    class WebRowDragEndEventArgsDetailDescription extends WebBaseEventArgsDetailDescription {
        constructor() {
            super();
            this.p = null;
            this.k = null;
            this.n = false;
        }
        get_type() {
            return "WebRowDragEndEventArgsDetail";
        }
        get dragDirective() {
            return this.p;
        }
        set dragDirective(a) {
            this.p = a;
            this.e("DragDirective");
        }
        get dragData() {
            return this.k;
        }
        set dragData(a) {
            this.k = a;
            this.e("DragData");
        }
        get animation() {
            return this.n;
        }
        set animation(a) {
            this.n = a;
            this.e("Animation");
        }
    }
    WebRowDragEndEventArgsDetailDescription.$t = /*@__PURE__*/ markType(WebRowDragEndEventArgsDetailDescription, 'WebRowDragEndEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebRowDragEndEventArgsDetailDescription.__marshalByValue1 = true;
    WebRowDragEndEventArgsDetailDescription.__marshalByValueAlias1 = "RowDragEndEventArgsDetail";
    return WebRowDragEndEventArgsDetailDescription;
})();
