// igniteui-react-core/esm2015/lib/WebSizeDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebSizeDescription = /*@__PURE__*/ (() => {
    class WebSizeDescription extends Description {
        constructor() {
            super();
            this.g = 0;
            this.f = 0;
        }
        get_type() {
            return "WebSize";
        }
        get type() {
            return this.get_type();
        }
        get width() {
            return this.g;
        }
        set width(a) {
            this.g = a;
            this.e("Width");
        }
        get height() {
            return this.f;
        }
        set height(a) {
            this.f = a;
            this.e("Height");
        }
    }
    WebSizeDescription.$t = /*@__PURE__*/ markType(WebSizeDescription, 'WebSizeDescription', Description.$);
    return WebSizeDescription;
})();
