// igniteui-react-core/esm2015/lib/WebSummaryResultDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebSummaryResultDescription = /*@__PURE__*/ (() => {
    class WebSummaryResultDescription extends Description {
        constructor() {
            super();
            this.l = null;
            this.m = null;
            this.i = null;
            this.g = false;
        }
        get_type() {
            return "WebSummaryResult";
        }
        get type() {
            return this.get_type();
        }
        get key() {
            return this.l;
        }
        set key(a) {
            this.l = a;
            this.e("Key");
        }
        get label() {
            return this.m;
        }
        set label(a) {
            this.m = a;
            this.e("Label");
        }
        get result() {
            return this.i;
        }
        set result(a) {
            this.i = a;
            this.e("Result");
        }
        get defaultFormatting() {
            return this.g;
        }
        set defaultFormatting(a) {
            this.g = a;
            this.e("DefaultFormatting");
        }
    }
    WebSummaryResultDescription.$t = /*@__PURE__*/ markType(WebSummaryResultDescription, 'WebSummaryResultDescription', Description.$);
    WebSummaryResultDescription.__marshalByValue = true;
    WebSummaryResultDescription.__marshalByValueAlias = "SummaryResult";
    return WebSummaryResultDescription;
})();
