// igniteui-react-core/esm2015/lib/WebSortingExpressionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebBaseEventArgsDescription } from "./WebBaseEventArgsDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebSortingExpressionDescription = /*@__PURE__*/ (() => {
    class WebSortingExpressionDescription extends WebBaseEventArgsDescription {
        constructor() {
            super();
            this.r = null;
            this.q = null;
            this.n = false;
            this.k = null;
        }
        get_type() {
            return "WebSortingExpression";
        }
        get fieldName() {
            return this.r;
        }
        set fieldName(a) {
            this.r = a;
            this.e("FieldName");
        }
        get dir() {
            return this.q;
        }
        set dir(a) {
            this.q = a;
            this.e("Dir");
        }
        get ignoreCase() {
            return this.n;
        }
        set ignoreCase(a) {
            this.n = a;
            this.e("IgnoreCase");
        }
        get strategy() {
            return this.k;
        }
        set strategy(a) {
            this.k = a;
            this.e("Strategy");
        }
    }
    WebSortingExpressionDescription.$t = /*@__PURE__*/ markType(WebSortingExpressionDescription, 'WebSortingExpressionDescription', WebBaseEventArgsDescription.$);
    WebSortingExpressionDescription.__marshalByValue1 = true;
    WebSortingExpressionDescription.__marshalByValueAlias1 = "SortingExpression";
    return WebSortingExpressionDescription;
})();
