// igniteui-react-core/esm5/lib/OrdinalTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { CategoryXAxisDescription } from "./CategoryXAxisDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var OrdinalTimeXAxisDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(OrdinalTimeXAxisDescription, _super);
    function OrdinalTimeXAxisDescription() {
        var _this = _super.call(this) || this;
        _this.el = null;
        _this.em = null;
        _this.ef = null;
        _this.ei = new Date();
        _this.eh = new Date();
        return _this;
    }
    OrdinalTimeXAxisDescription.prototype.get_type = function () {
        return "OrdinalTimeXAxis";
    };
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "dateTimeMemberPath", {
        get: function () {
            return this.el;
        },
        set: function (a) {
            this.el = a;
            this.e("DateTimeMemberPath");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "labellingMode", {
        get: function () {
            return this.em;
        },
        set: function (a) {
            this.em = a;
            this.e("LabellingMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "labelFormats", {
        get: function () {
            return this.ef;
        },
        set: function (a) {
            this.ef = a;
            this.e("LabelFormats");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "minimumValue", {
        get: function () {
            return this.ei;
        },
        set: function (a) {
            this.ei = a;
            this.e("MinimumValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "maximumValue", {
        get: function () {
            return this.eh;
        },
        set: function (a) {
            this.eh = a;
            this.e("MaximumValue");
        },
        enumerable: false,
        configurable: true
    });
    OrdinalTimeXAxisDescription.$t = markType(OrdinalTimeXAxisDescription, 'OrdinalTimeXAxisDescription', CategoryXAxisDescription.$);
    return OrdinalTimeXAxisDescription;
}(CategoryXAxisDescription));
export { OrdinalTimeXAxisDescription };
