// igniteui-react-core/esm5/lib/WebActionStripDescription_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebDisplayDensityBaseElementDescription } from "./WebDisplayDensityBaseElementDescription";
import { markType } from "./type";
import { Description } from "./Description";
/**
 * @hidden
 */
var WebActionStripDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebActionStripDescription, _super);
    function WebActionStripDescription() {
        var _this = _super.call(this) || this;
        _this.q = null;
        _this.k = null;
        _this.o = false;
        _this.m = null;
        _this.s = null;
        return _this;
    }
    WebActionStripDescription.prototype.get_type = function () {
        return "WebActionStrip";
    };
    Object.defineProperty(WebActionStripDescription.prototype, "context", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Context");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebActionStripDescription.prototype, "actionButtons", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("ActionButtons");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebActionStripDescription.prototype, "hidden", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Hidden");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebActionStripDescription.prototype, "resourceStrings", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("ResourceStrings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebActionStripDescription.prototype, "name", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    WebActionStripDescription.$t = markType(WebActionStripDescription, 'WebActionStripDescription', WebDisplayDensityBaseElementDescription.$);
    return WebActionStripDescription;
}(WebDisplayDensityBaseElementDescription));
export { WebActionStripDescription };
/**
 * @hidden
 */
var WebGridActionsBaseDirectiveDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridActionsBaseDirectiveDescription, _super);
    function WebGridActionsBaseDirectiveDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.j = false;
        _this.h = null;
        _this.l = null;
        return _this;
    }
    WebGridActionsBaseDirectiveDescription.prototype.get_type = function () {
        return "WebGridActionsBaseDirective";
    };
    Object.defineProperty(WebGridActionsBaseDirectiveDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionsBaseDirectiveDescription.prototype, "buttons", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Buttons");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionsBaseDirectiveDescription.prototype, "asMenuItems", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("AsMenuItems");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionsBaseDirectiveDescription.prototype, "strip", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Strip");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionsBaseDirectiveDescription.prototype, "name", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    WebGridActionsBaseDirectiveDescription.$t = markType(WebGridActionsBaseDirectiveDescription, 'WebGridActionsBaseDirectiveDescription', Description.$);
    return WebGridActionsBaseDirectiveDescription;
}(Description));
export { WebGridActionsBaseDirectiveDescription };
