// igniteui-react-core/esm5/lib/WebBaseToolbarDirectiveDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebGridToolbarBaseActionDescription } from "./WebGridToolbarBaseActionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebBaseToolbarDirectiveDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebBaseToolbarDirectiveDescription, _super);
    function WebBaseToolbarDirectiveDescription() {
        var _this = _super.call(this) || this;
        _this.m = null;
        _this.s = null;
        _this.r = null;
        _this.i = null;
        _this.o = null;
        _this.q = null;
        _this.p = null;
        _this.l = null;
        _this.k = null;
        _this.n = null;
        return _this;
    }
    WebBaseToolbarDirectiveDescription.prototype.get_type = function () {
        return "WebBaseToolbarDirective";
    };
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "columnListHeight", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("ColumnListHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "title", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Title");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "prompt", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("Prompt");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "overlaySettings", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("OverlaySettings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "name", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "openingRef", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("OpeningRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "openedRef", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("OpenedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "closingRef", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("ClosingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "closedRef", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("ClosedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebBaseToolbarDirectiveDescription.prototype, "columnToggleRef", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("ColumnToggleRef");
        },
        enumerable: false,
        configurable: true
    });
    WebBaseToolbarDirectiveDescription.$t = markType(WebBaseToolbarDirectiveDescription, 'WebBaseToolbarDirectiveDescription', WebGridToolbarBaseActionDescription.$);
    return WebBaseToolbarDirectiveDescription;
}(WebGridToolbarBaseActionDescription));
export { WebBaseToolbarDirectiveDescription };
