// igniteui-react-core/esm5/lib/WebCellDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebCellDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebCellDescription, _super);
    function WebCellDescription() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.i = null;
        _this.g = null;
        _this.f = null;
        return _this;
    }
    WebCellDescription.prototype.get_type = function () {
        return "WebCell";
    };
    Object.defineProperty(WebCellDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCellDescription.prototype, "primaryKey", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("PrimaryKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCellDescription.prototype, "state", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("State");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCellDescription.prototype, "pendingValue", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("PendingValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCellDescription.prototype, "editValue", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("EditValue");
        },
        enumerable: false,
        configurable: true
    });
    WebCellDescription.$t = markType(WebCellDescription, 'WebCellDescription', Description.$);
    return WebCellDescription;
}(Description));
export { WebCellDescription };
