// igniteui-react-core/esm5/lib/WebExporterEventDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebExporterEventDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebExporterEventDetailDescription, _super);
    function WebExporterEventDetailDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.h = null;
        _this.j = null;
        _this.m = false;
        return _this;
    }
    WebExporterEventDetailDescription.prototype.get_type = function () {
        return "WebExporterEventDetail";
    };
    Object.defineProperty(WebExporterEventDetailDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExporterEventDetailDescription.prototype, "exporter", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Exporter");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExporterEventDetailDescription.prototype, "options", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Options");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExporterEventDetailDescription.prototype, "grid", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Grid");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExporterEventDetailDescription.prototype, "cancel", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Cancel");
        },
        enumerable: false,
        configurable: true
    });
    WebExporterEventDetailDescription.$t = markType(WebExporterEventDetailDescription, 'WebExporterEventDetailDescription', Description.$);
    WebExporterEventDetailDescription.__marshalByValue = true;
    WebExporterEventDetailDescription.__marshalByValueAlias = "ExporterEventDetail";
    return WebExporterEventDetailDescription;
}(Description));
export { WebExporterEventDetailDescription };
