// igniteui-react-core/esm5/lib/WebFilteringOperationDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebFilteringOperationDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebFilteringOperationDescription, _super);
    function WebFilteringOperationDescription() {
        var _this = _super.call(this) || this;
        _this.m = null;
        _this.h = false;
        _this.l = null;
        _this.g = false;
        return _this;
    }
    WebFilteringOperationDescription.prototype.get_type = function () {
        return "WebFilteringOperation";
    };
    Object.defineProperty(WebFilteringOperationDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringOperationDescription.prototype, "name", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringOperationDescription.prototype, "isUnary", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("IsUnary");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringOperationDescription.prototype, "iconName", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IconName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringOperationDescription.prototype, "hidden", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Hidden");
        },
        enumerable: false,
        configurable: true
    });
    WebFilteringOperationDescription.$t = markType(WebFilteringOperationDescription, 'WebFilteringOperationDescription', Description.$);
    WebFilteringOperationDescription.__marshalByValue = true;
    WebFilteringOperationDescription.__marshalByValueAlias = "FilteringOperation";
    return WebFilteringOperationDescription;
}(Description));
export { WebFilteringOperationDescription };
