// igniteui-react-core/esm5/lib/WebFieldPipeArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebFieldPipeArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebFieldPipeArgsDescription, _super);
    function WebFieldPipeArgsDescription() {
        var _this = _super.call(this) || this;
        _this.k = null;
        _this.l = null;
        _this.i = null;
        _this.h = null;
        _this.j = null;
        _this.f = 0;
        return _this;
    }
    WebFieldPipeArgsDescription.prototype.get_type = function () {
        return "WebFieldPipeArgs";
    };
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "format", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Format");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "timezone", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Timezone");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "digitsInfo", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("DigitsInfo");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "currencyCode", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("CurrencyCode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "display", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Display");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFieldPipeArgsDescription.prototype, "weekStart", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("WeekStart");
        },
        enumerable: false,
        configurable: true
    });
    WebFieldPipeArgsDescription.$t = markType(WebFieldPipeArgsDescription, 'WebFieldPipeArgsDescription', Description.$);
    return WebFieldPipeArgsDescription;
}(Description));
export { WebFieldPipeArgsDescription };
