// igniteui-react-core/esm5/lib/WebFilteringExpressionsTreeDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebFilteringExpressionsTreeDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebFilteringExpressionsTreeDetailDescription, _super);
    function WebFilteringExpressionsTreeDetailDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.k = null;
        _this.j = null;
        _this.l = null;
        return _this;
    }
    WebFilteringExpressionsTreeDetailDescription.prototype.get_type = function () {
        return "WebFilteringExpressionsTreeDetail";
    };
    Object.defineProperty(WebFilteringExpressionsTreeDetailDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringExpressionsTreeDetailDescription.prototype, "filteringOperands", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("FilteringOperands");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringExpressionsTreeDetailDescription.prototype, "operator", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Operator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringExpressionsTreeDetailDescription.prototype, "fieldName", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("FieldName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFilteringExpressionsTreeDetailDescription.prototype, "treeType", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("TreeType");
        },
        enumerable: false,
        configurable: true
    });
    WebFilteringExpressionsTreeDetailDescription.$t = markType(WebFilteringExpressionsTreeDetailDescription, 'WebFilteringExpressionsTreeDetailDescription', Description.$);
    WebFilteringExpressionsTreeDetailDescription.__marshalByValue = true;
    WebFilteringExpressionsTreeDetailDescription.__marshalByValueAlias = "FilteringExpressionsTreeDetail";
    return WebFilteringExpressionsTreeDetailDescription;
}(Description));
export { WebFilteringExpressionsTreeDetailDescription };
