// igniteui-react-core/esm5/lib/WebGridSelectionRangeDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridSelectionRangeDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridSelectionRangeDescription, _super);
    function WebGridSelectionRangeDescription() {
        var _this = _super.call(this) || this;
        _this.g = 0;
        _this.f = 0;
        _this.k = null;
        _this.j = null;
        return _this;
    }
    WebGridSelectionRangeDescription.prototype.get_type = function () {
        return "WebGridSelectionRange";
    };
    Object.defineProperty(WebGridSelectionRangeDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDescription.prototype, "rowStart", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("RowStart");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDescription.prototype, "rowEnd", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("RowEnd");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDescription.prototype, "columnStart", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("ColumnStart");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDescription.prototype, "columnEnd", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("ColumnEnd");
        },
        enumerable: false,
        configurable: true
    });
    WebGridSelectionRangeDescription.$t = markType(WebGridSelectionRangeDescription, 'WebGridSelectionRangeDescription', Description.$);
    return WebGridSelectionRangeDescription;
}(Description));
export { WebGridSelectionRangeDescription };
