// igniteui-react-core/esm5/lib/WebGridToolbarExporterDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebBaseToolbarDirectiveDescription } from "./WebBaseToolbarDirectiveDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridToolbarExporterDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridToolbarExporterDescription, _super);
    function WebGridToolbarExporterDescription() {
        var _this = _super.call(this) || this;
        _this.ac = false;
        _this.ad = false;
        _this.ak = null;
        _this.ae = false;
        _this.aj = null;
        _this.ai = null;
        return _this;
    }
    WebGridToolbarExporterDescription.prototype.get_type = function () {
        return "WebGridToolbarExporter";
    };
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "exportCSV", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("ExportCSV");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "exportExcel", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("ExportExcel");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "filename", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("Filename");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "isExporting", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("IsExporting");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "exportStartedRef", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("ExportStartedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExporterDescription.prototype, "exportEndedRef", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("ExportEndedRef");
        },
        enumerable: false,
        configurable: true
    });
    WebGridToolbarExporterDescription.$t = markType(WebGridToolbarExporterDescription, 'WebGridToolbarExporterDescription', WebBaseToolbarDirectiveDescription.$);
    return WebGridToolbarExporterDescription;
}(WebBaseToolbarDirectiveDescription));
export { WebGridToolbarExporterDescription };
