// igniteui-react-core/esm5/lib/WebGridToolbarExportEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridToolbarExportEventArgsDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridToolbarExportEventArgsDetailDescription, _super);
    function WebGridToolbarExportEventArgsDetailDescription() {
        var _this = _super.call(this) || this;
        _this.o = null;
        _this.k = null;
        _this.m = null;
        _this.r = false;
        return _this;
    }
    WebGridToolbarExportEventArgsDetailDescription.prototype.get_type = function () {
        return "WebGridToolbarExportEventArgsDetail";
    };
    Object.defineProperty(WebGridToolbarExportEventArgsDetailDescription.prototype, "grid", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Grid");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExportEventArgsDetailDescription.prototype, "exporter", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Exporter");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExportEventArgsDetailDescription.prototype, "options", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Options");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridToolbarExportEventArgsDetailDescription.prototype, "cancel", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("Cancel");
        },
        enumerable: false,
        configurable: true
    });
    WebGridToolbarExportEventArgsDetailDescription.$t = markType(WebGridToolbarExportEventArgsDetailDescription, 'WebGridToolbarExportEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebGridToolbarExportEventArgsDetailDescription.__marshalByValue1 = true;
    WebGridToolbarExportEventArgsDetailDescription.__marshalByValueAlias1 = "GridToolbarExportEventArgsDetail";
    return WebGridToolbarExportEventArgsDetailDescription;
}(WebBaseEventArgsDetailDescription));
export { WebGridToolbarExportEventArgsDetailDescription };
