// igniteui-react-core/esm5/lib/WebGridActionButtonDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridActionButtonDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridActionButtonDescription, _super);
    function WebGridActionButtonDescription() {
        var _this = _super.call(this) || this;
        _this.f = false;
        _this.i = null;
        _this.h = null;
        _this.j = null;
        _this.k = null;
        return _this;
    }
    WebGridActionButtonDescription.prototype.get_type = function () {
        return "WebGridActionButton";
    };
    Object.defineProperty(WebGridActionButtonDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionButtonDescription.prototype, "asMenuItem", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("AsMenuItem");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionButtonDescription.prototype, "iconName", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("IconName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionButtonDescription.prototype, "classNames", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("ClassNames");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionButtonDescription.prototype, "iconSet", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("IconSet");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridActionButtonDescription.prototype, "labelText", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("LabelText");
        },
        enumerable: false,
        configurable: true
    });
    WebGridActionButtonDescription.$t = markType(WebGridActionButtonDescription, 'WebGridActionButtonDescription', Description.$);
    return WebGridActionButtonDescription;
}(Description));
export { WebGridActionButtonDescription };
