// igniteui-react-core/esm5/lib/WebGridEditDoneEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridEditDoneEventArgsDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridEditDoneEventArgsDetailDescription, _super);
    function WebGridEditDoneEventArgsDetailDescription() {
        var _this = _super.call(this) || this;
        _this.y = null;
        _this.o = null;
        _this.x = null;
        _this.w = null;
        _this.v = null;
        _this.k = null;
        _this.m = null;
        _this.r = false;
        _this.s = false;
        return _this;
    }
    WebGridEditDoneEventArgsDetailDescription.prototype.get_type = function () {
        return "WebGridEditDoneEventArgsDetail";
    };
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "rowID", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("RowID");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "cellID", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("CellID");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "rowData", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("RowData");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "oldValue", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("OldValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "newValue", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("NewValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "column", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Column");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "owner", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Owner");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "isAddRow", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("IsAddRow");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridEditDoneEventArgsDetailDescription.prototype, "valid", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Valid");
        },
        enumerable: false,
        configurable: true
    });
    WebGridEditDoneEventArgsDetailDescription.$t = markType(WebGridEditDoneEventArgsDetailDescription, 'WebGridEditDoneEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebGridEditDoneEventArgsDetailDescription.__marshalByValue1 = true;
    WebGridEditDoneEventArgsDetailDescription.__marshalByValueAlias1 = "GridEditDoneEventArgsDetail";
    return WebGridEditDoneEventArgsDetailDescription;
}(WebBaseEventArgsDetailDescription));
export { WebGridEditDoneEventArgsDetailDescription };
