// igniteui-react-core/esm5/lib/WebGridSelectionRangeDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebGridSelectionRangeDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebGridSelectionRangeDetailDescription, _super);
    function WebGridSelectionRangeDetailDescription() {
        var _this = _super.call(this) || this;
        _this.h = 0;
        _this.g = 0;
        _this.m = null;
        _this.l = null;
        return _this;
    }
    WebGridSelectionRangeDetailDescription.prototype.get_type = function () {
        return "WebGridSelectionRangeDetail";
    };
    Object.defineProperty(WebGridSelectionRangeDetailDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDetailDescription.prototype, "rowStart", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("RowStart");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDetailDescription.prototype, "rowEnd", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("RowEnd");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDetailDescription.prototype, "columnStart", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("ColumnStart");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebGridSelectionRangeDetailDescription.prototype, "columnEnd", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("ColumnEnd");
        },
        enumerable: false,
        configurable: true
    });
    WebGridSelectionRangeDetailDescription.$t = markType(WebGridSelectionRangeDetailDescription, 'WebGridSelectionRangeDetailDescription', Description.$);
    WebGridSelectionRangeDetailDescription.__marshalByValue = true;
    WebGridSelectionRangeDetailDescription.__marshalByValueAlias = "GridSelectionRangeDetail";
    return WebGridSelectionRangeDetailDescription;
}(Description));
export { WebGridSelectionRangeDetailDescription };
