// igniteui-react-core/esm5/lib/WebOverlaySettingsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebOverlaySettingsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebOverlaySettingsDescription, _super);
    function WebOverlaySettingsDescription() {
        var _this = _super.call(this) || this;
        _this.p = null;
        _this.f = null;
        _this.h = null;
        _this.l = false;
        _this.k = false;
        _this.j = false;
        return _this;
    }
    WebOverlaySettingsDescription.prototype.get_type = function () {
        return "WebOverlaySettings";
    };
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "target", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("Target");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "positionStrategy", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("PositionStrategy");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "scrollStrategy", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("ScrollStrategy");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "modal", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Modal");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "closeOnOutsideClick", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("CloseOnOutsideClick");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebOverlaySettingsDescription.prototype, "closeOnEscape", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("CloseOnEscape");
        },
        enumerable: false,
        configurable: true
    });
    WebOverlaySettingsDescription.$t = markType(WebOverlaySettingsDescription, 'WebOverlaySettingsDescription', Description.$);
    return WebOverlaySettingsDescription;
}(Description));
export { WebOverlaySettingsDescription };
