// igniteui-react-core/esm5/lib/WebPivotDateDimensionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebPivotDimensionDescription } from "./WebPivotDimensionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotDateDimensionDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotDateDimensionDescription, _super);
    function WebPivotDateDimensionDescription() {
        var _this = _super.call(this) || this;
        _this.ah = false;
        _this.ak = null;
        _this.aa = null;
        _this.ae = null;
        _this.ac = null;
        return _this;
    }
    WebPivotDateDimensionDescription.prototype.get_type = function () {
        return "WebPivotDateDimension";
    };
    Object.defineProperty(WebPivotDateDimensionDescription.prototype, "enabled", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("Enabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionDescription.prototype, "dataType", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("DataType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionDescription.prototype, "resourceStrings", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("ResourceStrings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionDescription.prototype, "inBaseDimension", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("InBaseDimension");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionDescription.prototype, "inOptions", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("InOptions");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotDateDimensionDescription.$t = markType(WebPivotDateDimensionDescription, 'WebPivotDateDimensionDescription', WebPivotDimensionDescription.$);
    WebPivotDateDimensionDescription.__marshalByValue1 = true;
    WebPivotDateDimensionDescription.__marshalByValueAlias1 = "PivotDateDimension";
    return WebPivotDateDimensionDescription;
}(WebPivotDimensionDescription));
export { WebPivotDateDimensionDescription };
