// igniteui-react-core/esm5/lib/WebPaginatorResourceStringsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPaginatorResourceStringsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPaginatorResourceStringsDescription, _super);
    function WebPaginatorResourceStringsDescription() {
        var _this = _super.call(this) || this;
        _this.g = null;
        _this.j = null;
        _this.f = null;
        _this.k = null;
        _this.h = null;
        _this.i = null;
        return _this;
    }
    WebPaginatorResourceStringsDescription.prototype.get_type = function () {
        return "WebPaginatorResourceStrings";
    };
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_label", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Igx_paginator_label");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_pager_text", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Igx_paginator_pager_text");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_first_page_button_text", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Igx_paginator_first_page_button_text");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_previous_page_button_text", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Igx_paginator_previous_page_button_text");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_last_page_button_text", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Igx_paginator_last_page_button_text");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorResourceStringsDescription.prototype, "igx_paginator_next_page_button_text", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Igx_paginator_next_page_button_text");
        },
        enumerable: false,
        configurable: true
    });
    WebPaginatorResourceStringsDescription.$t = markType(WebPaginatorResourceStringsDescription, 'WebPaginatorResourceStringsDescription', Description.$);
    return WebPaginatorResourceStringsDescription;
}(Description));
export { WebPaginatorResourceStringsDescription };
