// igniteui-react-core/esm5/lib/WebPivotDataSelectorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotDataSelectorDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotDataSelectorDescription, _super);
    function WebPivotDataSelectorDescription() {
        var _this = _super.call(this) || this;
        _this.f = false;
        _this.h = false;
        _this.g = false;
        _this.i = false;
        _this.n = null;
        return _this;
    }
    WebPivotDataSelectorDescription.prototype.get_type = function () {
        return "WebPivotDataSelector";
    };
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "columnsExpanded", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ColumnsExpanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "rowsExpanded", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("RowsExpanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "filtersExpanded", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("FiltersExpanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "valuesExpanded", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("ValuesExpanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDataSelectorDescription.prototype, "gridRef", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("GridRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotDataSelectorDescription.$t = markType(WebPivotDataSelectorDescription, 'WebPivotDataSelectorDescription', Description.$);
    return WebPivotDataSelectorDescription;
}(Description));
export { WebPivotDataSelectorDescription };
