// igniteui-react-core/esm5/lib/WebPivotAggregatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotAggregatorDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotAggregatorDescription, _super);
    function WebPivotAggregatorDescription() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.i = null;
        _this.g = null;
        _this.f = null;
        return _this;
    }
    WebPivotAggregatorDescription.prototype.get_type = function () {
        return "WebPivotAggregator";
    };
    Object.defineProperty(WebPivotAggregatorDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotAggregatorDescription.prototype, "key", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Key");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotAggregatorDescription.prototype, "label", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Label");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotAggregatorDescription.prototype, "aggregatorName", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("AggregatorName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotAggregatorDescription.prototype, "aggregatorRef", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("AggregatorRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotAggregatorDescription.$t = markType(WebPivotAggregatorDescription, 'WebPivotAggregatorDescription', Description.$);
    return WebPivotAggregatorDescription;
}(Description));
export { WebPivotAggregatorDescription };
