// igniteui-react-core/esm5/lib/WebPivotKeysDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotKeysDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotKeysDescription, _super);
    function WebPivotKeysDescription() {
        var _this = _super.call(this) || this;
        _this.g = null;
        _this.j = null;
        _this.f = null;
        _this.i = null;
        _this.h = null;
        _this.k = null;
        return _this;
    }
    WebPivotKeysDescription.prototype.get_type = function () {
        return "WebPivotKeys";
    };
    Object.defineProperty(WebPivotKeysDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "children", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Children");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "records", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Records");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "aggregations", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Aggregations");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "level", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Level");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "columnDimensionSeparator", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("ColumnDimensionSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotKeysDescription.prototype, "rowDimensionSeparator", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("RowDimensionSeparator");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotKeysDescription.$t = markType(WebPivotKeysDescription, 'WebPivotKeysDescription', Description.$);
    return WebPivotKeysDescription;
}(Description));
export { WebPivotKeysDescription };
