// igniteui-react-core/esm5/lib/WebPivotValueDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotValueDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotValueDescription, _super);
    function WebPivotValueDescription() {
        var _this = _super.call(this) || this;
        _this.q = null;
        _this.o = null;
        _this.h = null;
        _this.f = null;
        _this.j = false;
        _this.l = null;
        _this.n = null;
        _this.p = null;
        return _this;
    }
    WebPivotValueDescription.prototype.get_type = function () {
        return "WebPivotValue";
    };
    Object.defineProperty(WebPivotValueDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "member", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Member");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "displayName", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("DisplayName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "aggregate", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Aggregate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "aggregateList", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("AggregateList");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "enabled", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Enabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "styles", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Styles");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "dataType", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("DataType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDescription.prototype, "formatterRef", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("FormatterRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotValueDescription.$t = markType(WebPivotValueDescription, 'WebPivotValueDescription', Description.$);
    return WebPivotValueDescription;
}(Description));
export { WebPivotValueDescription };
