// igniteui-react-core/esm5/lib/WebPivotValueDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotValueDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotValueDetailDescription, _super);
    function WebPivotValueDetailDescription() {
        var _this = _super.call(this) || this;
        _this.s = null;
        _this.q = null;
        _this.h = null;
        _this.f = null;
        _this.k = false;
        _this.m = null;
        _this.p = null;
        _this.r = null;
        return _this;
    }
    WebPivotValueDetailDescription.prototype.get_type = function () {
        return "WebPivotValueDetail";
    };
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "member", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Member");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "displayName", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("DisplayName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "aggregate", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Aggregate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "aggregateList", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("AggregateList");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "enabled", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Enabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "styles", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Styles");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "dataType", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("DataType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotValueDetailDescription.prototype, "formatterRef", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("FormatterRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotValueDetailDescription.$t = markType(WebPivotValueDetailDescription, 'WebPivotValueDetailDescription', Description.$);
    WebPivotValueDetailDescription.__marshalByValue = true;
    WebPivotValueDetailDescription.__marshalByValueAlias = "PivotValueDetail";
    return WebPivotValueDetailDescription;
}(Description));
export { WebPivotValueDetailDescription };
