// igniteui-react-core/esm5/lib/WebPivotConfigurationDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotConfigurationDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotConfigurationDescription, _super);
    function WebPivotConfigurationDescription() {
        var _this = _super.call(this) || this;
        _this.o = null;
        _this.n = null;
        _this.h = null;
        _this.f = null;
        _this.i = null;
        _this.g = null;
        _this.r = null;
        return _this;
    }
    WebPivotConfigurationDescription.prototype.get_type = function () {
        return "WebPivotConfiguration";
    };
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "rowStrategy", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("RowStrategy");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "columnStrategy", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("ColumnStrategy");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "rows", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Rows");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "columns", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Columns");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "values", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Values");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "filters", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Filters");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotConfigurationDescription.prototype, "pivotKeys", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("PivotKeys");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotConfigurationDescription.$t = markType(WebPivotConfigurationDescription, 'WebPivotConfigurationDescription', Description.$);
    return WebPivotConfigurationDescription;
}(Description));
export { WebPivotConfigurationDescription };
