// igniteui-react-core/esm5/lib/WebPivotDateDimensionOptionsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotDateDimensionOptionsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotDateDimensionOptionsDescription, _super);
    function WebPivotDateDimensionOptionsDescription() {
        var _this = _super.call(this) || this;
        _this.i = false;
        _this.j = false;
        _this.h = false;
        _this.g = false;
        _this.f = false;
        return _this;
    }
    WebPivotDateDimensionOptionsDescription.prototype.get_type = function () {
        return "WebPivotDateDimensionOptions";
    };
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "total", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Total");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "years", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Years");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "quarters", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Quarters");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "months", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Months");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDateDimensionOptionsDescription.prototype, "fullDate", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("FullDate");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotDateDimensionOptionsDescription.$t = markType(WebPivotDateDimensionOptionsDescription, 'WebPivotDateDimensionOptionsDescription', Description.$);
    return WebPivotDateDimensionOptionsDescription;
}(Description));
export { WebPivotDateDimensionOptionsDescription };
