// igniteui-react-core/esm5/lib/WebPivotDimensionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPivotDimensionDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPivotDimensionDescription, _super);
    function WebPivotDimensionDescription() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.r = null;
        _this.k = false;
        _this.f = null;
        _this.s = null;
        _this.p = null;
        _this.t = null;
        _this.m = 0;
        _this.q = null;
        return _this;
    }
    WebPivotDimensionDescription.prototype.get_type = function () {
        return "WebPivotDimension";
    };
    Object.defineProperty(WebPivotDimensionDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "childLevel", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("ChildLevel");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "memberName", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("MemberName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "enabled", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Enabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "filter", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Filter");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "sortDirection", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("SortDirection");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "dataType", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("DataType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "width", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("Width");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "level", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Level");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPivotDimensionDescription.prototype, "memberFunctionRef", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("MemberFunctionRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPivotDimensionDescription.$t = markType(WebPivotDimensionDescription, 'WebPivotDimensionDescription', Description.$);
    WebPivotDimensionDescription.__marshalByValue = true;
    WebPivotDimensionDescription.__marshalByValueAlias = "PivotDimension";
    return WebPivotDimensionDescription;
}(Description));
export { WebPivotDimensionDescription };
