// igniteui-react-core/esm5/lib/WebPaginatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebDisplayDensityBaseElementDescription } from "./WebDisplayDensityBaseElementDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPaginatorDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPaginatorDescription, _super);
    function WebPaginatorDescription() {
        var _this = _super.call(this) || this;
        _this.s = 0;
        _this.q = 0;
        _this.r = 0;
        _this.t = 0;
        _this.k = null;
        _this.m = null;
        _this.o = null;
        _this.y = null;
        _this.ac = null;
        _this.z = null;
        _this.aa = null;
        _this.ab = null;
        return _this;
    }
    WebPaginatorDescription.prototype.get_type = function () {
        return "WebPaginator";
    };
    Object.defineProperty(WebPaginatorDescription.prototype, "totalPages", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("TotalPages");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "page", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Page");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "perPage", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("PerPage");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "totalRecords", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("TotalRecords");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "selectOptions", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("SelectOptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "overlaySettings", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("OverlaySettings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "resourceStrings", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("ResourceStrings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "name", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "perPageChangeRef", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("PerPageChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "pageChangeRef", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("PageChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "pagingRef", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("PagingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaginatorDescription.prototype, "pagingDoneRef", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
            this.e("PagingDoneRef");
        },
        enumerable: false,
        configurable: true
    });
    WebPaginatorDescription.$t = markType(WebPaginatorDescription, 'WebPaginatorDescription', WebDisplayDensityBaseElementDescription.$);
    return WebPaginatorDescription;
}(WebDisplayDensityBaseElementDescription));
export { WebPaginatorDescription };
