// igniteui-react-core/esm5/lib/WebRowSelectionEventArgsDetailDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebBaseEventArgsDetailDescription } from "./WebBaseEventArgsDetailDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebRowSelectionEventArgsDetailDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebRowSelectionEventArgsDetailDescription, _super);
    function WebRowSelectionEventArgsDetailDescription() {
        var _this = _super.call(this) || this;
        _this.q = null;
        _this.p = null;
        _this.o = null;
        _this.r = null;
        _this.l = false;
        return _this;
    }
    WebRowSelectionEventArgsDetailDescription.prototype.get_type = function () {
        return "WebRowSelectionEventArgsDetail";
    };
    Object.defineProperty(WebRowSelectionEventArgsDetailDescription.prototype, "oldSelectionRef", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("OldSelectionRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectionEventArgsDetailDescription.prototype, "newSelectionRef", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("NewSelectionRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectionEventArgsDetailDescription.prototype, "addedRef", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("AddedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectionEventArgsDetailDescription.prototype, "removedRef", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("RemovedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectionEventArgsDetailDescription.prototype, "allRowsSelected", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("AllRowsSelected");
        },
        enumerable: false,
        configurable: true
    });
    WebRowSelectionEventArgsDetailDescription.$t = markType(WebRowSelectionEventArgsDetailDescription, 'WebRowSelectionEventArgsDetailDescription', WebBaseEventArgsDetailDescription.$);
    WebRowSelectionEventArgsDetailDescription.__marshalByValue1 = true;
    WebRowSelectionEventArgsDetailDescription.__marshalByValueAlias1 = "RowSelectionEventArgsDetail";
    return WebRowSelectionEventArgsDetailDescription;
}(WebBaseEventArgsDetailDescription));
export { WebRowSelectionEventArgsDetailDescription };
