// igniteui-react-core/esm5/lib/WebRowSelectorTemplateDetailsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebRowSelectorTemplateDetailsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebRowSelectorTemplateDetailsDescription, _super);
    function WebRowSelectorTemplateDetailsDescription() {
        var _this = _super.call(this) || this;
        _this.h = 0;
        _this.k = null;
        _this.j = null;
        _this.f = false;
        return _this;
    }
    WebRowSelectorTemplateDetailsDescription.prototype.get_type = function () {
        return "WebRowSelectorTemplateDetails";
    };
    Object.defineProperty(WebRowSelectorTemplateDetailsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectorTemplateDetailsDescription.prototype, "index", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Index");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectorTemplateDetailsDescription.prototype, "rowID", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("RowID");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectorTemplateDetailsDescription.prototype, "key", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Key");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowSelectorTemplateDetailsDescription.prototype, "selected", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Selected");
        },
        enumerable: false,
        configurable: true
    });
    WebRowSelectorTemplateDetailsDescription.$t = markType(WebRowSelectorTemplateDetailsDescription, 'WebRowSelectorTemplateDetailsDescription', Description.$);
    return WebRowSelectorTemplateDetailsDescription;
}(Description));
export { WebRowSelectorTemplateDetailsDescription };
