// igniteui-react-core/esm5/lib/WebRowTypeDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebRowTypeDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebRowTypeDescription, _super);
    function WebRowTypeDescription() {
        var _this = _super.call(this) || this;
        _this.aq = 0;
        _this.ar = 0;
        _this.ab = false;
        _this.ac = false;
        _this.n = null;
        _this.av = null;
        _this.l = null;
        _this.au = null;
        _this.f = null;
        _this.w = false;
        _this.ad = false;
        _this.ae = false;
        _this.x = false;
        _this.v = false;
        _this.aa = false;
        _this.g = null;
        _this.p = null;
        _this.z = false;
        _this.r = null;
        _this.u = false;
        _this.y = false;
        _this.j = null;
        return _this;
    }
    WebRowTypeDescription.prototype.get_type = function () {
        return "WebRowType";
    };
    Object.defineProperty(WebRowTypeDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "index", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("Index");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "viewIndex", {
        get: function () {
            return this.ar;
        },
        set: function (a) {
            this.ar = a;
            this.e("ViewIndex");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "isGroupByRow", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
            this.e("IsGroupByRow");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "isSummaryRow", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("IsSummaryRow");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "groupRow", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("GroupRow");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "key", {
        get: function () {
            return this.av;
        },
        set: function (a) {
            this.av = a;
            this.e("Key");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "validation", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Validation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "data", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("Data");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "cells", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Cells");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "disabled", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "pinned", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("Pinned");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "selected", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("Selected");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "expanded", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("Expanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "deleted", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("Deleted");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "inEditMode", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("InEditMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "children", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Children");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "rowParent", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("RowParent");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "hasChildren", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("HasChildren");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "treeRow", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("TreeRow");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "addRowUI", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("AddRowUI");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "focused", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("Focused");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebRowTypeDescription.prototype, "grid", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Grid");
        },
        enumerable: false,
        configurable: true
    });
    WebRowTypeDescription.$t = markType(WebRowTypeDescription, 'WebRowTypeDescription', Description.$);
    WebRowTypeDescription.__marshalByValue = true;
    WebRowTypeDescription.__marshalByValueAlias = "RowType";
    return WebRowTypeDescription;
}(Description));
export { WebRowTypeDescription };
