// igniteui-react-core/esm5/lib/WebSearchInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebSearchInfoDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebSearchInfoDescription, _super);
    function WebSearchInfoDescription() {
        var _this = _super.call(this) || this;
        _this.m = null;
        _this.f = false;
        _this.g = false;
        _this.j = 0;
        _this.l = null;
        return _this;
    }
    WebSearchInfoDescription.prototype.get_type = function () {
        return "WebSearchInfo";
    };
    Object.defineProperty(WebSearchInfoDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSearchInfoDescription.prototype, "searchText", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("SearchText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSearchInfoDescription.prototype, "caseSensitive", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("CaseSensitive");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSearchInfoDescription.prototype, "exactMatch", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("ExactMatch");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSearchInfoDescription.prototype, "activeMatchIndex", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("ActiveMatchIndex");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSearchInfoDescription.prototype, "matchInfoCacheRef", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("MatchInfoCacheRef");
        },
        enumerable: false,
        configurable: true
    });
    WebSearchInfoDescription.$t = markType(WebSearchInfoDescription, 'WebSearchInfoDescription', Description.$);
    return WebSearchInfoDescription;
}(Description));
export { WebSearchInfoDescription };
