// igniteui-react-core/esm5/lib/WebTreeGridRecordDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebTreeGridRecordDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebTreeGridRecordDescription, _super);
    function WebTreeGridRecordDescription() {
        var _this = _super.call(this) || this;
        _this.r = null;
        _this.q = null;
        _this.f = null;
        _this.h = null;
        _this.o = 0;
        _this.l = false;
        _this.k = false;
        return _this;
    }
    WebTreeGridRecordDescription.prototype.get_type = function () {
        return "WebTreeGridRecord";
    };
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "key", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("Key");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "data", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Data");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "children", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Children");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "recordParent", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("RecordParent");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "level", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Level");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "isFilteredOutParent", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IsFilteredOutParent");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridRecordDescription.prototype, "expanded", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Expanded");
        },
        enumerable: false,
        configurable: true
    });
    WebTreeGridRecordDescription.$t = markType(WebTreeGridRecordDescription, 'WebTreeGridRecordDescription', Description.$);
    WebTreeGridRecordDescription.__marshalByValue = true;
    WebTreeGridRecordDescription.__marshalByValueAlias = "TreeGridRecord";
    return WebTreeGridRecordDescription;
}(Description));
export { WebTreeGridRecordDescription };
