// igniteui-react-core/esm5/lib/WebTreeGridDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebGridBaseDirectiveDescription } from "./WebGridBaseDirectiveDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebTreeGridDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebTreeGridDescription, _super);
    function WebTreeGridDescription() {
        var _this = _super.call(this) || this;
        _this.hf = null;
        _this.hh = null;
        _this.hi = null;
        _this.g9 = false;
        _this.hj = null;
        _this.g6 = null;
        _this.g5 = null;
        _this.hg = null;
        _this.hd = null;
        _this.hb = 0;
        _this.hk = null;
        return _this;
    }
    WebTreeGridDescription.prototype.get_type = function () {
        return "WebTreeGrid";
    };
    Object.defineProperty(WebTreeGridDescription.prototype, "childDataKey", {
        get: function () {
            return this.hf;
        },
        set: function (a) {
            this.hf = a;
            this.e("ChildDataKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "foreignKey", {
        get: function () {
            return this.hh;
        },
        set: function (a) {
            this.hh = a;
            this.e("ForeignKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "hasChildrenKey", {
        get: function () {
            return this.hi;
        },
        set: function (a) {
            this.hi = a;
            this.e("HasChildrenKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "cascadeOnDelete", {
        get: function () {
            return this.g9;
        },
        set: function (a) {
            this.g9 = a;
            this.e("CascadeOnDelete");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "id", {
        get: function () {
            return this.hj;
        },
        set: function (a) {
            this.hj = a;
            this.e("Id");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "rootRecords", {
        get: function () {
            return this.g6;
        },
        set: function (a) {
            this.g6 = a;
            this.e("RootRecords");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "processedRootRecords", {
        get: function () {
            return this.g5;
        },
        set: function (a) {
            this.g5 = a;
            this.e("ProcessedRootRecords");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "dataRef", {
        get: function () {
            return this.hg;
        },
        set: function (a) {
            this.hg = a;
            this.e("DataRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "filteredData", {
        get: function () {
            return this.hd;
        },
        set: function (a) {
            this.hd = a;
            this.e("FilteredData");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "expansionDepth", {
        get: function () {
            return this.hb;
        },
        set: function (a) {
            this.hb = a;
            this.e("ExpansionDepth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeGridDescription.prototype, "rowLoadingIndicatorTemplateRef", {
        get: function () {
            return this.hk;
        },
        set: function (a) {
            this.hk = a;
            this.e("RowLoadingIndicatorTemplateRef");
        },
        enumerable: false,
        configurable: true
    });
    WebTreeGridDescription.$t = markType(WebTreeGridDescription, 'WebTreeGridDescription', WebGridBaseDirectiveDescription.$);
    return WebTreeGridDescription;
}(WebGridBaseDirectiveDescription));
export { WebTreeGridDescription };
